As next generation sequencing technologies continue to mature and find applications across genomics, it has become clear that the scale and scope of generated data far exceeds our ability for manual interpretation. Machine learning has shown remarkable success in finding patterns in this data and generating biologically testable hypotheses. In this thesis, I develop and apply machine learning methods to infer directional migration from whole genome sequencing (WGS) data and identify shared and distinct regulatory programs in closely related cell types from chromatin accessibility assays. 

The decreasing cost and increasingly availability of WGS has allowed for a global profiling of genomic diversity. Understanding the genetic history of these individuals has been a longstanding goal of population genomics, however there has never been a method capable of inferring directional migration over time. I introduce a particle filter for demographic inference (SMC2) which uses a flexible sequential Monte Carlo sampler alongside variational Bayes to determine likely population size and migration histories from WGS. I apply this particle filter to large genetic diversity biobanks and uncover an abundance of migration from the ancestors of non-Africans into Africa between 40 and 70 thousand years ago. I show that latent directional migration has broader implications for the inference of population size in gold-standard approaches and explore the ramifications of this migration in the context of African pre-history.

At the same time, functional genomics is using NGS-based chromatin accessibility assays to uncover active regulatory pathways in disease-relevant cell types. I introduce a method, BLDA, which uses latent Dirichlet allocation to explain shared and distinct regulatory pathways between different cell types. I demonstrate the method's utility by recovering known regulatory biology in blood cell development. I then apply BLDA to understand cis-regulatory element usage in a treatment resistant leukemia caused by the MLL-AF4 oncoprotein. The results highlight a previously uncharacterized class of enhancer elements depleted in DOT1L-deposited H3 lysine 79 methylation and enriched for PAF1c binding. 

This thesis introduces and applies machine learning approaches that use NGS to solve pressing problems in functional and population genomics. 