\begin{savequote}[8cm]
  It's easier to be terrified by an enemy you admire.
  \qauthor{--- Frank Herbert, \textit{Dune}}
\end{savequote}

\chapter{Identifying co-accessible regulatory regions using topic modelling} \label{ch4}
%\chapter{Regulatory Program Topic Modelling for ATAC-seq}
%\chapter{Latent Dirichlet Allocation for the Unsupervised Discrimination of ATAC-seq Experiments}
%\chapter{Machine learning, or something like that...}

\minitoc

\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}

%\section{Motivation}

\section{Introduction} \label{ch4:intro}

The physical accessibility of \Glspl{cre} in part determines which and how many transcription factor proteins are able to bind to the \gls{dna}. On the other hand, the binding of certain transcription factors, especially pioneer factors, has the ability to alter the accessibility of the chromatin \cite{Zaret2011}. The complex and dynamic interplay between these opposing forces regulates the process of transcription, controlling the expression of genes in a dynamic and contextual way \cite{Minnoye2021, Klemm}. Recent methods have permitted the profiling of chromatin accessibility on a genome-wide scale, however the role of the physical compaction of the genome and specific \glspl{cre} remains a poorly understood predictor of cell identity outside of niche model systems \cite{Schulz2019}. Understanding the dynamics of chromatin accessibility both between and within cell systems is of relevance to understanding the effect of sequence mutations that disrupt \glspl{tfbs} and which alter the availability of the entire \gls{cre}. Additionally, a thorough catalog of relevant and distinct accessible elements within a pathological cell system would allow for improved prioritization of dysregulated transcription factors and their genomic consequences. In this chapter, I aim to improve the characterization of regulatory programs through modelling chromatin accessibility with high throughput sequencing. The resulting method, BLDA, represents a viable approach to identify key regions of accessible chromatin that are both shared between similar cell types and discriminatory of others. I show that this method has similar power to identify important regions when compared to single cell ATAC-seq, and demonstrate its use on the well-characterised developmental stages between \glspl{hsc} and mature erythrocytes. The results here demonstrate that topic modelling is a reliable method for general purpose discrimination of important accessible regions in arbitrary collections of bulk \gls{atac} experiments.

\subsection{Transcriptional regulation through chromatin accessibility} \label{ch3:chrom_acc}

Beginning with a single copy of the diploid genome, sequential cellular differentiation creates upwards of $\sim$40 trillion individual cells with unique functions across organ systems and functional niches \cite{Quinlan2010}. Each of these cells contains essentially the same genetic code yet performs entirely different roles in the body, indicating the presence of an immensely intricate regulatory process dictating how genes are expressed in certain cellular contexts. As explored in \Cref{intro:atac}, a portion of this cell type specific regulation of expression is explained by DNA binding proteins, yet where and how these proteins are able to bind is dictated by the physical accessibility of the underlying sequence \cite{Bell2011}. Within the nucleus, DNA is packaged into a highly compact and organized structure known as chromatin. The construction of chromatin involves wrapping molecules of DNA around histone proteins. Typically, 147 base pairs of DNA wrap around an octomer of histone proteins to form the fundamental subunit of chromatin called a nucleosome \cite{Zaret2011}. 
Nucleosomes can act as a barrier to RNA polymerase II mediated transcription as well as many transcription factors \cite{Tsompana2014}. 
The exception to this rule are pioneer factors, which bind instead to closed chromatin and recruit chromatin remodellers to alter its accessibility, priming it for functional activities and the binding of other transcription factors \cite{HW2017,Zaret2011}. 
%The density and positioning of nucleosomes along the sequence determines to a large degree the ability of macromolecules to bind to the sequence. 
Other factors such as the post-translational modifications of histone proteins and higher order organization of chromatin also contribute to functional chromatin accessibility \cite{Gates2017}. 
As a consequence of these facts, nucleosomes tend to be found at lower densities within active regulatory regions; this makes the relative accessibility of a region an indicator of its regulatory capacity \cite{Tsompana2014}. 

The accessible genome comprises between 2 and 3 percent of the actual sequence in humans, but represents approximately 90 percent of regions bound by \gls{tf} \cite{Thurman2012}. The accessibility of a \gls{tf} binding site does not in and of itself dictate that it will bind, but represents a regulatory capacity at the accessible site \cite{Tsompana2014}. \Glspl{tf} bind to sequence competitively with histones and other chromatin binding proteins to dynamically modulate the organization and placement of nucleosomes \cite{Calo2013a}. While closed chromatin is generally not amenable to binding by \glspl{tf} other than the previously described pioneer factors, permissive chromatin is sufficiently dynamic to allow for some binding \cite{Bell2011a}. Accessibility therefore exists on a continuum that is dynamically reorganized in part based on the cellular context and its battery of expressed transcription factors.  Profiling the accessible regions of the genome in a particular cellular context provides a window into active regulatory programs. 

% Recent evidence suggests that chromatin accessibility is intricately involved in regulation of this process

%Moreso, due to the relatively short lifespan of mature blood cells, \glspl{hsc} are essential throughout life to provide multilineage progenitors for the many hematopoietic niches . This makes hematopoiesis an ideal system to further our understanding of stem cell biology, intricately linked to oncogenesis. Changes in chromatin accessibility are crucial to directing the progression of blood cell development \cite{Hu2016}. 

Several methods exist to experimentally determine chromatin accessibility. The majority of recent methods use an enzymatic reaction to selectively fragment the genomic sequence and next-generation sequencing to comprehensively survey the genome for the enrichment of fragments. Examples of these approaches include DNAse-seq, \gls{atac}, MNase-seq, FAIRE-seq, and NOMe-seq, reviewed in \textcite{Klemm} and \textcite{Meyer2014}. Of these, DNAse-seq is the most sensitive but requires a large number of cells to generate reliable libraries for sequencing \cite{Boyle2008}. ATAC-seq is a reliable method for characterising accessibility, applicable to systems with as few as 500 cells for bulk, or even on a per-cell basis \cite{Corces2017,Buenrostro2015}. Datasets generated with ATAC-seq have been growing exponentially year over year, representing in 2019 several times more data generated than any other approach for assaying chromatin accessibility \cite{Yan2020}. Thus, methods for the interpretation of large compendiums of ATAC-seq form a useful complement to the method's growing adaptation. 

\subsection{Regulation of key stages within hematopoiesis and erythropoiesis} \label{ch4:ery}

Of the $\sim$40 trillion cells in a typical human body, approximately 90\% derive from the hematopoietic lineage \cite{Quinlan2010}. The process of differentiation from \glspl{hsc} to mature blood cells is therefore of immense importance to health and disease. In adults, hematopoiesis begins from \glspl{hsc} which exist in a relatively quiescent state within the bone marrow, maintaining the capability for self-renewal and multipotenency \cite{Baron2012}. Because this differentiation process occurs throughout life, hematopoiesis represents an ideal model system to understand the properties of stem cell biology, the dysregulation of which is intricately linked with oncogenesis \cite{Orkin2008}. These cells give rise to all other phenotypically distinct blood cells through a hierarchical cascade of differentiation (\Cref{fig:hem_sum}). Several intermediary points in this process are relevant to this chapter, including the initial priming of the stem cells into multipotent progenitor cells (MPPs). This involves, among other things, active polycomb repression of lineage specification genes such as \textit{Ebf1} and \textit{Pax5} in MPP cells \cite{H2010}. Lineage commitment represents a decision point, where continuing specification will exclusively occur within a specific branch of the hematopoietic differentiation tree. In this chapter, we use erythropoiesis as a model system to biologically validate the model we introduce. 

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/hem.pdf}
  \caption[Hematopoiesis schematic]{Summary of important cell types arising during hematopoiesis from HSCs to phenotypically and functionally distinct cell types. Adapted from \textcite{Hu2016}. HSC = hematopoietic stem cell.}
  \label{fig:hem_sum}
\end{figure}

The regulation of chromatin accessibility plays a central role in the differentiation trajectory of erythrocytes \cite{Hu2016,Schulz2019}. Erythropoiesis refers to the process of  successive differentiation from pluripotent stem and progenitor cells through several morphologically and functionally distinct stages to form enucleated erythrocytes (also known as red blood cells). Here we focus on the process up until erythroblasts, the penultimate step in erythropoiesis and the last before enucleation. \textcite{Ludwig2019} used FACS sorting on CD71, CD235a, CD49d, and BAND3 surface markers to identify eight distinct stages of development. Their analysis using both ATAC-seq and RNA-seq is the most complete representation of this differentiation trajectory to date. The authors identify myeloid progenitors (MyP), colony forming units - erythroid (CFU-E), pro-erythoblasts with two stages (ProE1, ProE2), basophilic erythoblasts (Baso-E), polychromatic erythroblasts (Poly-E), orthochromatic erythoblasts (OrthoE), and orthochromatic reticulocyte (Ortho/Ret) as key cell stages. A detailed explanation of the molecular biology of each stage is not the focus of this thesis, however interested readers may consult texts such as \textcite{Sinclair2013}. After augmenting their dataset with hematopoietic progenitor cells created by \textcite{Corces2016}, the authors found that accessible elements clustered into several groupings. Some were predominantly active in the earliest stages of hematopoiesis, while others were broadly accessible across lineage commitment and intermediate erythropoiesis, while others still acted primarily in terminal erythropoiesis. 
These broad groupings allowed the authors to identify several factors of interest which may act differentially; we investigate these factors such as \textit{TMCC2, UROS}, and \textit{RHAG} as well as other well known markers like \textit{TAL1, KLF5} and \textit{GATA1} later in this chapter. In this chapter, we recreate this dataset by augmenting the eight erythroid cell type presented in \textcite{Ludwig2019} with the hematopoietic stem and progenitor cells from \textcite{Corces2016} to create a dataset of chromatin accessibility throughout erythropoiesis.


\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/ery}
  \caption[Erythropoiesis schematic]{Key stages of erythropoiesis from \textcite{Sinclair2013}. Blast forming units are not represented in the \textcite{Ludwig2019} dataset, while erythroblasts replace erythrocytes as the end point of the process for our purposes.}
  \label{fig:ery}
\end{figure}


\subsection{Identifying regulatory programs in large databases}

The task of identifying co-accessible sets of regulatory elements usually falls on differential peak accessibility analyses \cite{Yan2020}. Many methods exist for this task, including MACS2, DiffBind, csaw, voom, limma, edgeR, and DESeq2 \cite{Ritchie2015,Law2014,Love2014,Robinson2009,StarkRandBrown2016,Lun2015,Zhang2008}. A comprehensive review of their relative performances was undertaken by \textcite{Reske2020}. In general, these methods work by grouping similar experiments, usually taken to be biological replicates, and finding peak regions whose deviation overcomes some level of statistical significance. These tools are highly refined and benchmarked for applications with two cell types of interest. However, using differential accessibility in large systems with multiple consistent subpopulations poses technical issues. As an example, edgeR is able to estimate coefficients for a general linear model and find significant differences in accessibility for a particular cluster. However, the estimation of statistical significance relies on estimating variance with biological replicates, and the user is warned not to attempt significance testing without performing replicates \cite{Robinson2009}. Though this places no hard constraint on the data to be analysed, clustering in large datasets is imperfect. The estimation of the variance is therefore critically reliant on the homogeneity of the cluster, which may vary between clusters. This makes the interpretation of differentially accessible elements between clusters in large datasets difficult. Secondly, it is difficult to use differential accessibility testing to find patterns unique to combinations of clusters, as the estimated coefficients tend to relate to enrichment in a specific cluster, or between all of them. One alternative is to look at the pairwise differential accessibility between clusters and search for patterns in the identified regions. This is, however, onerous and to our knowledge no dedicated method exists for the task.  

Recently, \textcite{BravoGonzalez-Blas2019} proposed the use of topic modelling to study collections of accessible regions in single cell \gls{scatac}. Topic modelling, specifically \gls{lda} represents a viable method for the unsupervised identification of key regions of accessible sequence in large databases, while simultaneously identifying the distribution of their associated regulatory programs. The method learns groupings of differentially accessible elements and where they tend to be active, in aggregate. However, the use of LDA in large collections of ATAC-seq data has not been explored. As the amount of available ATAC-seq data grows year over year, analyses of large compendiums of cellular variation in accessibility form a useful complement to sparse single cell analyses. On the one hand, single cell analyses are well powered to identify fine-scale groupings of regulatory elements active in sub-populations of similar cell types. On the other, analysis of dense bulk ATAC-seq with high read depths allows for a thorough investigation of pathways active across large groupings of cells. In this chapter, we investigate the use of LDA for identifying regulatory programs and their distribution in bulk \gls{atac}.

\subsection{Latent Dirichlet Allocation}

A detailed methodological motivation of \gls{lda} is beyond the scope of this chapter, however here I present in general terms a typical formulation of the generative model and inference from real data.  

\subsubsection{The generative model}

LDA is interested in the generation of a corpus of $D$ documents, composed of $n$ words. It was introduced in \textcite{Blei2003}. In brief, each document $d = 1 \ldots D$ can be described as a collection of weights on $k$ topics. An example in the realm of natural language processing would be books and their respective genres (i.e. fiction, science, etc.). In the specific application here presented, the corpus represents the dataset of sequencing experiments, with each document representing a single ATAC-seq sample. The words are individual accessible regions, which are determined and discretized through peak calling on respective samples. The inference procedure is unsupervised, and there is no annotation or metadata given to discriminate amongst samples, unlike later adaptations like structured topic modelling \cite{Roberts2019}.

The generative process is as follows. For a fixed number of topics $k$, each document $d$ is given a distribution on its topic weights $\theta_d$ $$ \theta_d \sim Dir(\alpha), d = 1, \ldots, D $$ Each word $n$ in document $d$ is generated from a particular topic $z_{dn}$ where  $z_{dn}$ $$ z_{dn} \sim Discrete(\theta_d) $$ and the word itself is chosen from $$ p(w_{dn} | z_{dn}, \zeta) $$ which is a multinomial probability conditional on the topic allocation $z_{dn}$. The other dependency is on $\zeta$, here defined as a $k \times N$ matrix where $\zeta_{ij} = P(w_j =1 | z_i = 1)$, or the conditional probability of a word given a topic allocation. This is a fixed quantity that will be estimated. 

The Dirichlet distribution is convenient in this case, as draws from $Dir(\alpha)$ represent points on the ($k-1$) simplex where parameter $\alpha$ is a $k$-vector of positive reals. A simplex is a multidimensional generalization of the triangle. The parameter vector $\alpha$ is also known as the concentration parameter which, in the case of symmetric Dirichlet distributions where $\alpha_i = \alpha_j \forall i,j \in 1 \ldots k$ determines the sparsity of the draws from the distribution. In the limit of extremely high positive values of the concentration parameter, draws from the distribution tend to be approximately uniform draws over the probability space, while at values less than one and approaching zero the distribution is degenerate \cite{Ferguson1991}.  In addition, the Dirichlet distribution is the conjugate prior to the multinomial distribution, a fact which is exploited for efficient inference algorithms. Explicitly, the probability density function of the Dirichlet distribution here considered is given by 

$$ p(\theta | \alpha ) = \frac{\Gamma(\sum^k_{i=1} \alpha_i)}{ \prod^k_{i=1} \Gamma(\alpha_i)} \theta_1^{\alpha_1 -1} \ldots \theta_k^{\alpha_k-1} $$ which leads to an overall joint distribution and probability of an overall corpus as given in \textcite{Blei2003}.

This traditional set up for LDA does not allow for any control over the sparsity of word allocations to topics. For that, an extension called Smoothed LDA replaces the $\zeta$ matrix of conditional word probabilities with a second Dirichlet distribution, parameterized by $\beta$. As a side note, all the applications of LDA in this chapter are concerned with symmetric Dirichlet distributions, where each of the entries of the $\alpha$ and $\beta$ $k$-vectors are the same. This modified generative procedure proceeds identically, except that a seperate draw is made for each topic representing the distribution on its words from $$ \psi_k \sim Dir(\beta), k = 1, \ldots, K $$, and drawing a word conditional on its topic assignment is instead generated as $$w_{dn} \sim Discrete( \psi_{z_{dn}})$$ where $\psi_{z_{dn}}$ intuitively represents the topic-word distribution. In the case here considered, it represents the importance of a particular region of accessible chromatin to a regulatory program $z$.

\subsubsection{Parameter inference}

The joint distribution of latent parameters $\theta, z$ conditional on $w, \alpha, \beta$ does not admit analytical inference as the posterior probability is intractable, as noted in \textcite{Blei2003}. They proposed an approach based on variational Bayesian inference, however it has become commonplace to integrate out $\theta$ and approach the problem of inference using collapsed Gibbs sampling \cite{Qiu2014,Magnusson2018,Park2019}. Gibbs sampling is an extension of the \gls{mcmc} approach for sampling from the posterior distribution when, as is the case here, analytical inference is not possible. Gibbs sampling is closely related to the Metropolis-Hastings algorithm, and relies on constructing a Markov chain with the previously described posterior as its equilibrium distribution. In this way, inference about parameters is deduced from direct samples of the posterior. This is philosophically very similar to the approach taken in Chapter 3, where samples of the posterior distribution of genealogies conditioned on mutations were used to draw inference about demographic parameters such as directional migration. The details of the inference procedure are not of direct relevance to the results presented in this chapter, and a more detailed description can be found in \textcite{Qiu2014} and others.

Here, the posterior distribution of interest concerns the most likely values of $\theta$ and $\psi$, that is the topic weight vectors on each of the different ATAC-seq experiments and the topic distribution on each of the constituent accessible regions, conditioned on observed regions within ATAC-seq experiments and given parameters $\alpha, \beta$.

\subsection{The LDA algorithm and cisTopic}

The basis of this chapter is the cisTopic algorithm, and some expansion on its basic formulation is necessary. cisTopic is introduced in \textcite{BravoGonzalez-Blas2019}, and is primarily intended for use on \gls{scatac} data resulting from a combination of sequencing data and associated peak calls. In general, this data is either encoded in a count matrix or it is internally converted to one. A count matrix $M$, for the purposes of this chapter, refers to a $C \times R$ matrix where element $M_{cr}$ equals the number of reads (or fragments) overlapping region $r$ in cell $c$, for regions $r=1,\ldots,R$ and cells $c=1,\ldots,C$. Each region $r$ is selected on the basis of statistical peak calls, typically performed with software such as Macs2 or similar, on the aggregated \gls{scatac} signal. This count matrix $M$ is then subjected to binarization on the basis of some threshold $T$, where $T$ is the minimum number of counts necessary for a region to be declared accessible. This threshold reflects the low sequencing depth of \gls{scatac} data, and the difficulty in comparing cells quantitatively based on read counts alone without some correction for depth. This threshold is set to 1 by default in the stable version of the algorithm implementation. It is necessary to provide symmetric hyperparameters $\alpha$ and $\beta$, which cisTopic prefers to normalize by the total number of topics and provide an $\alpha$-per-topic value rather than an $\alpha$-per-sey value with which to do inference. We follow this convention in this thesis, providing unnormalized $\alpha$ values whenever relevant and raw $\beta$ values.   

The corrected count matrix is used for the inference of the cell-topic distribution and the region-topic distribution, previously represented by $\theta$ and $\psi$ respectively as the distribution of topics over documents and the distribution of words over topics. Topic loadings for both the cell and region loadings are normalized to the range [0, 1]. Key regions which are important to the topic may be selected in one of two ways. Firstly, by fitting a gamma distribution to the normalized region-topic loadings on a per-topic basis and selecting a percent point threshold of the resulting density's tail (i.e. the top 1 percentile of the fit gamma distribution). Alternatively, a given number of top regions may be selected based on the rank of the region-topic loadings.  

Selecting the right number of topics for a particular analysis is not straight forward. cisTopic implements an optimized version of collapsed Gibbs sampling by using WarpLDA, an algorithm for constant time inference \cite{Chen2016a}. An advantage of WarpLDA is that it returns the second derivative of each value for $k$. As log-likelihood values increase with increasing $k$, cisTopic makes use of these values to automatically select a value for $k$ based on a range given by the user. A proof that this procedure produces optimal values of $k$ is not readily available. 

\subsection{Aims of this chapter}

The overarching aim of this chapter is to investigate the use of LDA for bulk \gls{atac}. As noted above, the method represents a significant advancement above differential accessibility, and shows theoretical promise for the analysis of the growing collections of sequencing data in diverse cell systems. In doing so, I adapt the existing cisTopic method for bulk samples. Specifically, I investigate the performance of bulk LDA when compared to established \gls{scatac} inference using cisTopic. I also investigate whether the method infers meaningful topic loadings on a well understood system with ground truth values to compare against, erythropoiesis.   

% What is the goal. 
%   Identify important discriminatory regulatory elements and how they are shared amongst groups of cells. 
% Why LDA?
%   differential peak analysis can decide a statistical threshold for two different cell types.
%   there are many different methods for DA analyses MACS2, DiffBind, csaw, voom, limma, edgeR, and DESeq2 [27,28,29,30,31,32,33]
%   it is sensitive to the normalisation and data processing, given how these interact with the statistic model of choice (https://epigeneticsandchromatin.biomedcentral.com/articles/10.1186/s13072-020-00342-y)
%   Impractical to perform pairwise comparisons and attempt to infer patterns in this by eye.
%   Some approaches exist to look at pathways of cells (citation here??? ) during differentation. However, the problem of identifying both common and discriminatory regulatory elements remains open. 
%   Recently LDA has been proposed to look at the accessibility of dna in single cell atac-seq.

%   Advantages:
%     no assumption that a region is important only to a certain cell type or topic. A single region can contribute to many different topics, in different contexts.
%     consider groupings of elements as important, gives annotation to grouops of regions more so than answering the question of "what it different", it is answering, what is important 

%   Disadvantages:
%     Bag of words assumes no direct interaction betewen regions. Though there is biological rationale to consider how regions interact with each other, it is difficult to formulate this beyond co-occurance.

% Discussion should include somethign about there being more comprehensive models of topic modelling like structured topic modelling (stm package and https://cbail.github.io/SICSS_Topic_Modeling.html)

\section{Methods} \label{ch4:methods}


\subsection{Single Cell ATAC-seq Dataset Generation} \label{methods:sc_ds}

A single cell ATAC-seq dataset was compiled from data generated in \textcite{Buenrostro2015} for three cell types: K562, GM12878, and H1ESC. These cell types were selected as a subset which represents maximal diversity in accessible chromatin within the larger dataset. 

For each labelled cell type, we collected accession numbers within the NCBI sequence read archive. These include records SRR1780163 through SRR1780354 (K562), SRR1779683 through SRR1779778 (GM12878), and SRR1779589 through SRR1779683 (H1ESC). Sequencing reads were merged and adapters were trimmed with cutadapt v 2.10 using the following adapter sequences ({\tt -a CTGTCTCTTATACACATCT -A CTGTCTCTTATACACATCT}) \cite{Martin2011}. Quality of the merged dataset was verified with fastqc and aligned to hg19 using bowtie2 \cite{Andrews2010, Langmead2013}. Cell specific barcodes were added to the resulting alignment file within the {\tt CB} tag using pySam \cite{Heger2009}. Peak calling was performed using MACS2 and LanceOTron using default parameters \cite{Gaspar2018, Hentges2021} as detailed in \autoref{ch4:method_peaks} and count matrices were constructed as detailed in \autoref{ch4:method_cistopic}. A more thorough discussion of our choice to compare MACS2 and LanceOTron may be found in \Cref{intro:pc}.

\subsection{Construction of Pseudo-bulk ATAC-seq Dataset}

In order to construct a pseudo-bulk dataset, entries for each of the single cells were merged into a single alignment file. Cell barcodes were replaced with a cell-type marker and peak calling was similarly conducted with macs2 and LanceOTron \cite{Gaspar2018, Hentges2021}.

\subsection{Peak Calling from Coverage Data} \label{ch4:method_peaks}

We compare two methods of peak calling.

A public implementation of the LanceOTron can be found at \url{https://github.com/chris1221/lanceotron}. LanceOTron relies on a pre-trained neural network. We use the supplied weights in \url{wide_and_deep_fully_trained_v5_03.h5} along with the standard scaler values from the same implementation. We allow the network to identify candidate peaks and assign a peak score, thresholding our selected peaks on a peak score of at least 0.5 as in the implementation of LanceOTron at \url{https://lanceotron.molbiol.ox.ac.uk/}.


\subsection{Running LDA with cisTopic} \label{ch4:method_cistopic}

We used the implementation of \gls{lda} in cisTopic \cite{BravoGonzalez-Blas2019}. cisTopic is intended for use on single cell ATAC-seq experiments, however the input format is amenable to any quantitative data observed on a cell by region basis. To construct the input data to cisTopic, we create a bespoke pipeline that firstly calls peaks from ATAC-seq experiments, and secondly harmonizes these peaks while constructing a count matrix. A count matrix is a sparse matrix where the rows are cells, or in the case of this thesis, cell types in the form of ATAC-seq samples, and the columns are individual peak regions.

\subsection{Bayesian Hyper-parameter Optimization} \label{ch4:hyper}

\gls{lda} requires a set of three hyperparameters to be supplied, alpha, beta, and the number of topics. This chapter investigates applications of LDA to problems of several scales, so the choice of hyperparameters is not easily chosen, at least in an unbiased way. We approach this issue by using a Bayesian Optimization approach to learn the best set of hyperparameters for a given set of ATAC-seq experiments. We wrote a typical cisTopic analysis in python via rpy2 ({\tt https://rpy2.github.io/doc/latest/html/index.html}) and used the BayesianOptimization library ({\tt https://github.com/fmfn/BayesianOptimization}) to optimize a target function for a given set of hyperparameters. This task was facilitated by the use of Dask to compute several hundred possible combinations simultaneously \cite{Rocklin2015}. The target function to optimise is based on the specific application, as discussed in \autoref{ch4:results} section.

\subsection{Bulk LDA (BLDA) Method} \label{ch4:method_blda}

The BLDA method is a small extension to cisTopic which aims to incorporate a proxy of how accessible each peak region is in a particular experiment. This makes it in some ways more similar to the traditional approach to LDA within natural language processing, which gives an integer value for the number of times a particular word appears in a text. To create the RPKM normalized count matrix and run the inference, we use functions from the {\tt blda} python package available on Github at \url{https://github.com/Chris1221/blda}. The construction of the count matrix requires linked coverage files and their associated peak calls. The pipeline is agnostic to the type of coverage file (either BAM files or RPKM normalised bigWig tracks can be provided) and peak calls can be made by any software which outputs delineated, non-overlapping genomic regions. Hyperparameter optimization is conducted according to \Cref{ch4:hyper} and a modified version of cisTopic is run using {\tt rpy2}. The modification to cisTopic, found at \url{https://github.com/Chris1221/cisTopic}, relaxes a constraint to provide the {\tt lda} package's collapsed Gibb's sampler with a binary sparse matrix. Several other functions from the cisTopic function are applicable directly to the situation of bulk LDA after the data has been altered. 

\subsection{Computing the average fold enrichment of topics for groups} \label{methods:average_fc}

To get a measure for the number of topics to use in a particular model, we compute the average fold enrichment. This is done with a two-step process. First, a one-tailed student's T test is performed for each of the grouping of cells. The cells are split into one of the cell types, with the remainder forming the comparison group. The null hypothesis here is that the means do not differ, or that the comparison group is significantly smaller than the group under question. The P values are corrected for the number of comparisons using Bonferroni correction (multiplying by the number of effective tests, which in this case is the number of tests) and the significant associations are reported. Second, the differences in the estimated means are divided to form a fold change metric amongst the subset of significant topic associations. We take the median of these values to form the average fold change metric for a particular set of topics being modelled. The median is selected to avoid extreme results overly biasing the statistic.

\subsection{Co-enrichment of topics}

The number of times two topics $i$ and $j$ are co-enriched is defined to be the count of occurrences where a single cell type has a topic loading for $i$ and $j$ exceeding some pre-defined threshold. After some experimentation, we decide to use a threshold of 0.25, where the two co-enriched topics collectively form half of the normalised topic loadings. 

\section{Results} \label{ch4:results}

LDA has previously been shown to be highly effective at detecting true patterns in shared accessibility in single cell data. Here, I create a dataset of known cell types in order to set baseline expectations about the capabilities of LDA in a single cell system before moving on to studying bulk samples.

\subsection{Bulk LDA recapitulates patterns from single cell ATAC-seq}

I generated a dataset of three labelled cell types, as per \Cref{methods:sc_ds}. This dataset comprises three cell types which have substantial differences in their accessible DNA.  K562 is an erythro-leukemic cell model of chronic myelogenous leukemia from a fifty-three-year old female donor \cite{Lozzio1975}. GM12878 is an \gls{ebv} immortalized B-lymphocyte cell line cultured from a female member of the international HapMap project CEPH panel \cite{Belmont2005}. Finally, H1ESC are a human embryonic stem cell line collected from a healthy male donor of unknown age \cite{Thomson1998}. 

The dataset comprises 96 GM12878 cells with a total of 1928090 reads, 96 H1ESC cells with 4192100 reads in total, and 192 K562 cells from two biological replicates with a total of 5269388 reads.

We generate coverage tracks for the merged single cell dataset using deepTools and peaks are called using both MACS2 and LanceOTron as per \Cref{ch4:method_peaks} \cite{Ramirez2014}. MACS2 identifies 10,075 statistically significant peak regions, while MACS2 finds 12,994, 8337 of which are common between the two approaches.  

\subsubsection{Identifying differentially accessible peaks with EdgeR} \label{ch4:edgeR}

The R package {\tt edgeR} is used to set a baseline expectation for differentially accessible regions of the genome between the clusters. {\tt edgeR} is selected as it was recently shown to have the overall highest sensitivity for identifying differential peaks from properly normalized ATAC-seq data. We took the single cell dataset and created a count matrix with all true counts across different cells. We inputted these into edgeR and estimated dispersion coefficients as recommended in the tutorial, fitting a generalized linear model according to the true cell identities which assume are known in this case. We identify the top 100 differentially accessible regions based on this analysis, and use these to study the results from our topic modelling approaches.

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/sim_ll_surface.pdf}
  \slcaption{Hyperparameter log-likelihood surface for two hyperparameters. Alpha and beta parameters are given as symmetric hyperparameters of the Dirichlet distribution, where alpha is normalized according to the number of topics.}
  \label{fig:llhood_surfacea_simulation}
\end{figure}

We investigate topic modelling in three separate scenarios. Firstly, we ran cisTopic as intended, using the single cell data. Secondly, we created pseudo-bulk alignment files for each of the clusters and run cisTopic normally. Thirdly, we adapted cisTopic to accept normalized read count. This allowed us to validate our baseline assumption that LDA is able to capture similar information within bulk sequencing experiments.

\subsubsection{LDA captures realistic regulatory patterns in known single cell systems} \label{ch4:sc}

For the first case, we model the system of 384 cells with 4, 5, 6, 7, and 8 topics to assess the best biological fit to the system. For each pre-set number of topics, hyperparameters are selected using Bayesian optimization. The best value for $\alpha$ was not well estimated, with considerable noise and no clear peak; this is in contrast to the $\beta$ value, which showed higher log-likelihoods at higher values approaching 1 (\Cref{fig:sc_opt_params}). Consistent with this, a high value is selected for $\beta$ in each case. $\beta$ controls the number of regions loaded to a topic. With a number of cells much larger than the number of topics, especially in the present scenario, a high value of $\beta$ represents a prior on more regions loaded to each of the small number of topics. This indicates that a relatively high number of regions are differentially accessible between the different clusters. The relatively uncertain distribution of alpha log-likelihoods represents an ambiguity in the likelihood from different Dirichlet distributions. Draws with a low alpha tend to represent topic loadings which are specific to individual cells, and accordingly the flat likelihood surface may indicate that there are multiple ways of assigning the topics to cells with relatively equal likelihoods.

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/sc_opt_params}
  \slcaption{Single cell log likelihood for different values of the topic modelling hyperparameters $\alpha$ and $\beta$ for various numbers of topics being modelled. Alpha and beta parameters are given as symmetric hyperparameters of the Dirichlet distribution, where alpha is normalized according to the number of topics}
  \label{fig:sc_opt_params}
\end{figure}

We investigate the loading of these topics on each of the cells in the collection (\Cref{fig:sc_topics}). As the number of topics modelled increases, some cell systems split more readily than others. H1ESC, for instance, has two enriched topics in even the smallest four topic model. For each of the number of topics, we calculate the median fold enrichment (\Cref{methods:average_fc}) and find that five and six topics produce very similar median fold enrichments (5.81 versus 5.82 respectively). To reduce the complexity of the model, and acknowledging that a model with more parameters will tend to fit data better, we choose the study the case with five topics.  

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/sc_topics.pdf}
  \slcaption{Topic loadings for 4, 5, 6, 7, and 8 topic instances of the LDA inference procedure using optimal hyperparamters as decided in \Cref{fig:sc_opt_params}. Inset shows average fold change amongst enriched topics for each of the number of topics modelled. Enriched topics are identified by a one-tailed student's $t$ test for difference between a known cell cluster and the remainder of cells.} 
  \label{fig:sc_topics}
\end{figure}

\paragraph{Choice of peak caller influences the inference of topic loadings}

We additionally investigate the role that peak calling has on the inference of topics. 

Peaks were called with Macs2 instead of LanceOTRon, resulting in a larger number of peaks identified in general, with 4657 being uniquely identified by Macs2 (\Cref{fig:sc_macs2}A). We repeated the above analysis, including the hyperparameter search, topic inference, and calculation of median fold enrichment was conducted using MACS2 instead of LanceOTron, as detailed in \Cref{ch4:method_peaks}. Several differences are apparent in both the optimized hyperparameters and the inferred topic loadings (\Cref{fig:sc_macs2}). Slight bias towards lower $\alpha$ values is observed in this dataset, and the same trend towards preferring higher values of $\beta$ is replicated here \Cref{fig:sc_macs2}B, C. However, when looking at the inferred topics, we find lower median fold enrichment in general (\Cref{fig:sc_macs2}D, E). From this we conclude that LanceOTron has identified a more specific set of accessible regions for the purposes of this study, and use it in all analyses going forward. 

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/sc_macs2.pdf}
  \slcaption{Analysis of single cell dataset with peaks called by MACS2. A. The number of peak calls made by both callers, with their overlapping portion displayed in the center of the Venn diagram. B. Likelihood surface for the optimization of the $\alpha$ hyperparameter. C. Likelihood surface for the optimization of the $\beta$ hyperparameter. D. Median fold enrichment for topics inferred under optimal hyperparameters. E. Inferred topic loadings on single cells using optimal hyperparameters.}
  \label{fig:sc_macs2}
\end{figure}

LDA produces both a distribution of topic loadings on cells as well as topic loadings on regions. In order to identify regions putatively identified with the cell types through the topic loadings, a decision must be made about how to discretise the continuous loadings to a set of enriched regions. The cisTopic package accomplishes this by fitting a gamma distribution to the topic loadings, and identifying regions within the top user-defined percentile of this distribution. By default, this is set to the top 5\% of the distribution. Here we investigate how to set this parameter value and the influence it has on the resulting post-hoc analyses. 

\paragraph{Enrichment of differentially accessible regions within identified topics}

To assess whether the identified topic loadings represent similar trends to the baseline differential accessibility analysis, we fit a gamma distribution to the shape of each topic loading and take the top 2.5 percentile of the distribution as representative so-called "keywords" or key regions. We use bedtools to overlap these key regions with the 100 differentially accessible regions from the {\tt edgeR} analysis and find that 61 of the 100 regions are shared with the top 415 selected key regions (\Cref{table:sc_t5_over}). Given that there are 10074 regions in total, this represents a significant ($P<0.00001$, $\chi^2 = 827.33$) enrichment of differentially expressed regions in the selected regions from the LDA procedure.


\begin{table}
  \centering
  \begin{tabular}{l|l|l}
  Topic & Total Selected Regions & Intersecting Regions  \\ 
  \hline
  1     & 104                    & 15                    \\
  2     & 3                      & 0                     \\
  3     & 194                    & 29                    \\
  4     & 3                      & 1                     \\
  5     & 111                    & 19                   
  \end{tabular}
  \slcaption{Five topic single cell inferred topic loadings and the proportion of their selected regions which overlap 100 established differentially accessible regions. Overall, 61 of the 100 regions were found in the top 415 regions.}
  \label{table:sc_t5_over}
  \end{table}

\subsubsection{Extending LDA to Pseudo-bulked Single Cells}

We investigate whether the approach implemented in cisTopic is appropriate for use in bulk ATAC-seq experiments. This change represents a deviation in the intended system for the approach. Practically, the difference between the single cell case and the bulk one is the difference between a wide, sparse count matrix and a dense one with fewer observations. An attractive property of using Gibb's sampling to infer the posterior cell-topic distribution is its ability to efficiently deal with sparsity in the region-topic distribution. However, it is not clear to what degree this sparsity is a requirement for the procedure to obtain biologically relevant topics, each representing a proxy of co-accessible regulatory elements. In this section, I take the well-characterized single cell dataset from the previous section and study the analogous bulk sequencing case by artificially creating bulk samples from the individual known cell types. We denote these new bulk cells as pseudobulked samples.

We combine all reads from each known cell type into pseudo-bulked alignment files using pySam. Peak calling is performed separately on each of the pseudo-bulked samples using LanceOTron, as described in the single cell case. We additionally explore the role of thresholding on LanceOTRon peak calls by creating a subset of calls with at least 0.5 peak score. This is a recommendation made by the web interface to the peak caller. Before thresholding, 517620, 395953, and 455886 peaks were identified in the H1ESC, GM12878, and K562 cells respectively. After merging, 1063153 regions were used for the complete analysis. After thresholding on a peak score of 0.5, 7524, 4945, and 6722 peaks were found in the GM12878, H1ESC and K562 cells respectively. After merging, there were 16210 regions in total used for the thresholded case. 

We take two approaches to constructing the count matrix. The first is the already described cisTopic method, where peak regions are merged and annotated by contributing cell types, leading to a binary sparse matrix where the entries $i, j$ represent overlapped peak $j$ in cell type $i$. We denote this case as "one-hot encoded" as the representation is analogous to one-hot encoding factor levels within a design matrix. This approach is justified with the single cell case as read depth is much lower per cell, and a quantification of relative read support for a particular peak region is highly variable. In the case of bulk ATAC-seq however, a difference in read support for different regions can imply varying degrees of accessibility, a key feature of closely related cell types or differentiation processes. To reflect this in the model, we propose an extension of the cisTopic method which we call bulk LDA (BLDA). In this extension, we normalize read counts according to the floor of their reads per kilobase pair and million reads (RPKM) normalized value, giving an integer value of effective number of times the "word", or region, is represented in the particular dataset. This approach has the advantage of incorporating quantitative information about differential peak strengths across experiments, and more closely mimics typical applications of LDA within natural language processing. Here we investigate whether the quantitative extension of cisTopic for bulk ATAC-seq is able to better recapitulate the single cell case.

\begin{table}
  \centering
  \begin{tabular}[t]{lrrr}
  \toprule
  Method & Number of topics & Best Alpha & Best Beta\\
  \midrule
  \cellcolor{gray!6}{One-hot encoding} & \cellcolor{gray!6}{4} & \cellcolor{gray!6}{123.68822} & \cellcolor{gray!6}{0.0382351}\\
  One-hot encoding & 5 & 466.58752 & 0.0665157\\
  \cellcolor{gray!6}{One-hot encoding} & \cellcolor{gray!6}{6} & \cellcolor{gray!6}{450.96735} & \cellcolor{gray!6}{0.0756824}\\
  One-hot encoding & 7 & 337.58572 & 0.0769945\\
  \cellcolor{gray!6}{One-hot encoding} & \cellcolor{gray!6}{8} & \cellcolor{gray!6}{100.35205} & \cellcolor{gray!6}{0.0872716}\\
  \addlinespace
  RPKM Normalization & 4 & 40.38505 & 0.2350854\\
  \cellcolor{gray!6}{RPKM Normalization} & \cellcolor{gray!6}{5} & \cellcolor{gray!6}{285.51986} & \cellcolor{gray!6}{0.3011584}\\
  RPKM Normalization & 6 & 285.70995 & 0.4163319\\
  \cellcolor{gray!6}{RPKM Normalization} & \cellcolor{gray!6}{7} & \cellcolor{gray!6}{366.02309} & \cellcolor{gray!6}{0.3825605}\\
  RPKM Normalization & 8 & 101.26750 & 0.3455916\\
  \bottomrule
  \end{tabular}
  \slcaption{Optimal LDA hyperparameters for pseudo-bulked scATAC-seq parameterized by \textit{a priori} defined topic numbers and two different read quantification methods.}
  \label{table:pb_opt_params}
\end{table}

We decided hyperparameters through Bayesian Optimization with four through eight topics, the same as the single cell case for consistency for both approaches separately. We have omitted the likelihood surfaces and provided the optimal parameters after 500 iterations in \Cref{table:pb_opt_params}. We run classic LDA for the \gls{ohe} and our extended BLDA cases using the selected hyperparameters and compare the inferred topic loadings between the approaches (\Cref{fig:pb_no_thresh_lot_topics}). Given the small number of dense cells which share a high proportion of peak regions, in both the entire peak dataset and the thresholded one the BLDA method produces sharper and more defined topic loadings onto the individual pseudo-bulk cells. The cell-topic distribution is not noticeably different between the thresholded and non-thresholded peak calls (A versus B, \Cref{fig:pb_no_thresh_lot_topics}).
The OHE case for this data was unable to produce topics which differentiated between the difference cell types, while the RPKM normalization may have allowed the inference procedure to select regions which varied in their accessibility (left versus right, \Cref{fig:pb_no_thresh_lot_topics}). 

\begin{figure}
  \centering
  \includegraphics[width = \textwidth]{plot/ch4/pb_thresholding_topics.pdf}
  \slcaption{Inferred topic loadings using optimized hyperparameters and \textit{a priori} defined for both the OHE and BLDA methods using pseudo-bulked scATAC-seq data.}
  \label{fig:pb_no_thresh_lot_topics}
\end{figure}

To investigate to what degree the BLDA procedure found regions which were differentially accessible across pseudo cell-types, we compare the "key word" regions to the regions selected by edgeR in \Cref{ch4:edgeR}. We use cisTopic's built in procedure for selecting important contributory regions to topic definitions, fitting a gamma distribution to the shape of the region-topic distribution for each topic and selecting regions which lie in the top 1\% of that distribution. Doing this, we find a difference in the number of regions identified between the thresholded and non-thresholded LanceOTron peak calls, which we expect given the selection procedure (\Cref{fig:pb_tot_number}). More peaks are also uniformly selected for the BLDA method, though we expect that this is due to the lack convergence in the classic LDA set up, rather than a systematic difference based on the method. 

\begin{figure}
  \centering
  \includegraphics[]{plot/ch4/pb_diff_acc_tot.pdf}
  \slcaption{Total number of selected "key-word" regions for a given topic number between the two pseudo-bulk approaches, the two peak calling methods (thresholded and non-thresholded LanceOTron) and single cell analyses.}

  \label{fig:pb_tot_number}
\end{figure}

However, we find that the key-word regions are not equally specific with regards to the "ground-truth" differentially accessible regions from edgeR. Taking the top 100 regions from the edgeR analysis, we overlap them with all key-word regions in a given analysis (\Cref{fig:pb_olap_number}). 

Firstly, we find a clear difference between OHE and BLDA in their ability to identify truly differentially accessible regions (First versus second panel of \Cref{fig:pb_olap_number}). Considering the case of thresholded peak calls, OHE identifies almost none of the known differentially accessible peaks, while the BLDA method identifies slightly fewer than the original single cell experiment. This supports the assertion that bulk LDA is a comparable approach to single cell LDA for datasets consisting of a small number of dense cells. 

\begin{figure}
  \centering
  \includegraphics[]{plot/ch4/pb_diff_acc_olap.pdf}
  \slcaption{Number of overlapping regions between the top 100 differentially accessible regions determined by EdgeR and key-word regions selected by taking the top 1\% of a fitted gamma distribution for several LDA analyses. }
  \label{fig:pb_olap_number}
\end{figure}

\subsection{Bulk LDA describes Erythropoiesis}

Having established that BLDA is able to identify realistic patterns in bulk data, we investigate a well-characterized biological system. Erythropoiesis, the process by which red blood cells are produced, involves known differentiation stages with defined marker genes. This allows us to compare the topics inferred from BLDA with expectations based on the biology of the system.

To create a dataset of bulk ATAC-seq data to study erythropoiesis, we download and process raw sequencing data from \textcite{Ludwig2019} and \textcite{Corces2016} using gene expression omnibus datasets GSE115684 and GSE74912 respectively. Raw reads were assessed for quality with FastQC and aligned using. Coverage tracks were created using DeepTools and peak calling was performed using LanceOTron using the same score cutoff as previously described. The number of peaks identified this way varied considerably across cell types (\Cref{fig:ludwig_shared_peaks}A). Accessible regions of the genome generally increase from hematopoietic stem cells (HSCs) to common myeloid progenitors before decreasing in Megakaryocyte–erythroid progenitor cell. Erythrocyte colony forming units (CFUE) and orthochromatic erythroblasts (OrthoE cells) have especially accessible chromatin, with nearly 50,000 accessible regions. As it is known that nuclear chromatin condenses in preparation for enucleation in terminally differentiated immature erythroblasts, accesibility and the number of identified peak regions are greatly decreased in erythroblasts here as well \cite{Schulz2019}. There is relatively low sharing of accessible regions outside of a central differentiation "block" between ProE and BasoE cells (\Cref{fig:ludwig_shared_peaks}B). This too is consistent with our expectation, as terminal differentiation is known to significantly alter the expression of many hundreds of genes \cite{Schulz2019}. It is now well understood to what degree the differential usage versus differential accessibility of key regulatory elements shapes normal differentiation of human cells \cite{Song2011,West2014}. Here we use the previously characterized bulk LDA to identify groupings of accessible regions which are associated with specific stages in erythropoiesis. 


\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/ludwig_shared_peaks.pdf}
  \slcaption{Peak calling in erythropoiesis dataset. A. Number of identified peak regions by cell type using LanceOTron and a score threshold of 0.5. B. The number of peaks shared between cell types.}
  \label{fig:ludwig_shared_peaks}
\end{figure}

Peak calls were merged to create two count matrices. The first represents the typical cisTopic analysis method, one-hot encoding each peak region from its derivative cell type. The second is our BLDA method, using an integer value of RPKM normalised read count for each identified accessible region. We optimise the hyperparameters as before, and run inference for between 8 and 20 topics. The number of topics was chosen to demonstrate, on the lower end, how LDA will group similar cells when it is forced to, and on the upper end if there is any remaining structure after each cell is allowed its own topic. We find that, similar to pseudo-bulked scATAC-seq data, the BLDA method consistently produces cleaner and better defined topic loadings for individual cell types (\Cref{fig:ludwig_quick_topics}). While this larger and more realistic dataset allows the OHE strategy to pick out some topics which differentiate between cell types, the patterns tend to be strongly constrained to certain grouping such as the erythroid precursors or lineage committed cells. The BLDA method on the other hand distinguishes highly cell-specific topics. The number of topics which are shared across multiple cell types varies considerably as the model is given more freedom with increasing number of topics. Even at low numbers, highly differentiated cell types like CD4 and CD8 T cells show distinct, highly enriched topics. Conversely, topics are almost always identified which are enriched in closely related intermediary cell types. We quantify the number of times two cell types share an enriched topic, taking a cell-normalised topic loading threshold of 0.25 to indicate enrichment (\Cref{fig:ludwig_quick_coenriched}). Two blocks of co-enriched cell types are obvious, one beginning from erythroid progenitors (ProE) and ending with basophilic erythroblasts (BasoE), the second beginning with BasoE and ending with Erythroblasts (Ery). It is rare for a cell type in either one of these clusters to share topic loadings with the other. Additionally, topics are occasionally shared between sequential stages of early differentiation, i.e. between hematopoietic stem cells and multi-potent progenitors (MPPs), but these topics never overlap with lineage committed cell stages.  


\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/ludwig_quick_topics.pdf}
  \slcaption{Inferred topic loadings for erythropoiesis dataset. On the left hand side, One-hot encoded LDA, on the right bulk LDA with RPKM normalization. Facets indicate the number of modelled topics. Cells are indicated in reverse differentiation pseudotime, with each facet ordered on the Y axis as CD8, CD4, CLP, LMPP, Erythroblast, Ortho/Ret, OrthoE, PolyE, BasoE, ProE2, ProE1, CFUE, ProE, MEP, CMP, MPP, and HSC. See \Cref{fig:ludwig_8_topic} for a detailed exploration for $k=8$.}
  \label{fig:ludwig_quick_topics}
\end{figure}

\begin{figure}
  \centering
  \includegraphics[width=0.5\textwidth]{plot/ch4/ludwig_coenriched.pdf}
  \slcaption{Similarity of cell types based on the number of times they are co-enriched for a topic, summarized over all numbers of inferred topics. Topics inferred using BLDA for the erythropoiesis dataset as described in the Methods section. For a description of the cell types, see \Cref{ch4:ery}}
  \label{fig:ludwig_quick_coenriched}
\end{figure}

\subsubsection{Isolating key-word regions from region-topic loadings}

Certain regions are inferred to be uniquely important for a topic. We follow the convention within the LDA literature in calling these samples key-words, as the observations within a document are typically denoted as words. There is no theoretical definition of a key-word region, therefore a threshold on the region-topic distribution must be empirically chosen to represent the most important regions. The cisTopic method normalizes the distribution of loadings within a topic such that it falls in the range 0-1 and theoretically follows a Gamma distribution, though the parameters for this distribution must again be inferred empirically. In this section, we attempt to identify sensible thresholds for key-word regions based on topic loadings. 

To decrease the number of variables that need to be studied, for this section we will focus on the BLDA inference. Based on the qualitative results from the previous section, quantitative input to the LDA inference algorithm produces more specific topics which are shared amongst related cell types in realistic ways. This specificity is important for the identifications of key-word regions. 

Additionally, we begin our analysis by focusing on only one value for the number of topics. For the analysis with $k = 16$ topics, SciPy was used to estimate the parameters of a Gamma distribution for the region-topic distribution for each of the 16 topics. Four thresholds were used to select the top 2.5, 1, 0.1, and 0.01\% of the inferred Gamma distribution. The number of selected regions ranged considerably across the topics (\Cref{fig:ludwig_topics_gamma}). From this we conclude that a gamma distribution may not fit each of the topics equally well, as the proportion of peaks that were selected based on the threshold does not reflect the theoretical expectation. 

To investigate further, descriptive statistics are used to identify candidate distributions. We use a Cullen a Frey graph to examine one of the poorly fit topics from \Cref{fig:ludwig_topics_gamma}, topic 8 (\Cref{fig:ludwig_topic8_distr}). 1000 bootstraps of the data indicate two potential distributions, gamma and log-normal (top left of \Cref{fig:ludwig_topics_gamma}). Maximum likelihood is used to estimate parameters for these distributions, and the empirical versus theoretical density, percentiles, and cumulative distribution functions are plotted (top right, bottom left, and bottom right of \Cref{fig:ludwig_topics_gamma} respectively).  Though it appears as though the density at least visually matches the gamma distribution, the larger empirical percentiles of the data do not match either a gamma or log-normal distribution (bottom left of \Cref{fig:ludwig_topic8_distr}). It seems likely that the skew of the data is causing the under-representation of selected key-word regions. 


\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/ludwig_topics_gamma_16.pdf}
  \slcaption{Number of identified regions above the facetted percent point function of a Gamma distribution with inferred parameters using SciPy.}
  \label{fig:ludwig_topics_gamma}
\end{figure}

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/topic8_distribution.pdf}
  \slcaption{Distribution of the region-topic distribution from a candidate topic from \Cref{fig:ludwig_topics_gamma}, topic 8. Top left shows a comparison of 1000 bootstrapped values of the data against descriptive statistics for several common distributions. Top right shows the empirical histogram of the data compared against the two distributions nominated from the Cullen and Frey graph fitted via maximum likelihood estimation. Top left shows empirical versus theoretical percentiles from the two fitted distributions, while bottom right shows the empirical cumulative distribution function.}
  \label{fig:ludwig_topic8_distr}
\end{figure}

This lack of fit to a theoretical gamma distribution presents several avenues for exploration. One option would be to fit a mixture distribution, explaining different portions of the data with different parameters for the distributions. However, this would make it difficult to estimate a certain proportion of the overall distribution, which is the goal of the exercise. Instead, we opt to take the simpler route and simply take a fixed number of the highest region-normalised topic loadings. This guarantees a sufficient sample size for further analyses like motif identification and pathway enrichment, while not making the analysis overly complex for replication across the different $k$ values of topics.

\subsubsection{Motif enrichment within key-word regions}

The top 100, 250, and 500 regions from each region-topic distribution are selected and MotifScan (\url{https://motifscan.readthedocs.io/en/latest/}) is used along with the JASPAR database of \gls{tfbs} motifs to identify regions matching known \gls{pwm} and find over-representation in the set of regions. A control set of regions is constructed by taking the negative intersection between the original merged dataset of regions and the selected regions, in order to better represent both accessible DNA and relevant common motifs for the system. A second control set is constructed by taking the union of all regions selected as key-word regions. This second set investigates motifs which are enriched relative to other selected regions.

The smallest $k$ topics forces sharing between similar cell types (\Cref{fig:ludwig_8_topic}). For each topic, the enrichment of all \gls{pwm} in the JASPAR database is calculated for each count of top regions, and each control strategy (\Cref{fig:ludwig_motifs_8}). Some topics in this particular instance are of interest. Topic 6 spans the early developmental stages between hematopoietic stem cells and multi-potent progenitors, and shows enrichment for known factors like MEIS1, whose expression is required for erythropoiesis in HSCs \cite{Miller2016,Zeddies2014,Unnisa2012}.  

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/ludwig_quick_topic_8.pdf}
  \slcaption{Cell-topic distribution for $k=8$ topic. A. Loadings across cell types ordered by differentiation trajectory. B. Loadings by topic across differentiation pseudo-time.}
  \label{fig:ludwig_8_topic}
\end{figure}

\begin{figure}
  \centering
  \includegraphics[width=\textwidth]{plot/ch4/ludwig_motifs_8.png}
  \slcaption{All identified motifs for $k=8$ topic analysis. The top 100, 250, and 500 topics were selected and enrichment was calculated either against all peak regions, or against all selected key-word regions. }
  \label{fig:ludwig_motifs_8}
\end{figure}

\subsubsection{BLDA identifies relevant pathways active in Erythropoiesis}

A combination of the motifs identified as well as the groupings of regions allows for a deeper interrogation of the uncovered biology.

\textcite{Ludwig2019} identified several regions of the genome with substantial impacts on the process of Erythropoiesis. Of these, there are no overlaps between UROS, CCND3, VEGFA, and TMCC2 in the set of selected keyword regions. However, the promoter region for the Rh associated glycoprotein (RhAG), which is not accessible in HSC, MPP, or CMP cells, shows a progressive increase in accessibility throughout erythropoiesis, as shown in \Cref{fig:RhAG}. A keyword region for topic 2 is located in its promoter. Topic 2 mirrors the accessibility patterns of RhAG, showing  which also is shown to be inactive in early hematopoiesis and increasing in accessible after lineage commitment, peaking in BasoE cells. Other examples of individual regions which mirror topic loading patterns are readily apparent, such as the R3HDM4 locus, which is also a key region for topic 2 (\Cref{fig:r3hdm4}). 

To comprehensively survey the closest annotated protein coding genes, bedtools was used to find the nearest genic regions to each of the key regions for the different topics. \textcite{Mello2019} recently surveyed differentially accessible genes throughout different phases of erythoiesis. These genes are compared against the nearest genic regions for the top 500 keyword regions from each of the topics. The results illustrate a close match between topic activation patterns and the underlying biology of blood cell differentiation.

Topic 1, active in early lineage commitment, include critical proteins for erythropoiesis such as STAT5A and STK3. STAT5a is predominantly expressed in early erythropoiesis \cite{Pishesha2014}. The regions are enriched for the binding motifs of GATA2, GATA3, KLF1, and KLF12. GATA3 is typically associated with lymphoid precursors and committed T cells, recent results indicate a more ubiquotous role within hematopoiesis \cite{Chen2001}. The topic shows no overlap with genes involved in apoptosis, unlike later topics.  

Topic 2 represents the main inferred mid-to-late erythropoiesis grouping of regulatory elements. It contains the majority of important proteins involved in iron homeostasis and mitochondrial transportation (SLC25A38), heme production (ALAS2), regulation of cellular differentiation (SP1). In addition, SP1 represents the first protein involved with enucleation. An gene set enrichment analysis showed subenrichment for "Positive regulation of erythrocyte differentiation", indicating that a significant portion of the genes identified relate to erythropoiesis. Additional terms identified include "tetrapyrrole biosynthetic process", "tetrapyrrole metabolic process", and "pigment metabolic process". \todo{Table for the GO enrichment with panther} 

Topic 3 peaks in activity amongst orthochromatic erythroblasts, and also has some representation in early hematopoiesis, as well as committed T cells. Topic 3 includes NRF1, a knockout of which leads to embryonic lethality due to impaired fetal liver erythropoiesis \cite{Chen2001}. GLRX5, predominantly expressed in late erythropoiesis by poly and orthochromatic erythroblasts (Fig 1c \cite{Pishesha2014}) is also amongst the nearest key genes, along with FOXO3 and PPP2R1A \cite{Mello2019}. Topic 3 also shows enrichment for a large number of motifs, indicative that a number of identified peak regions are not directly involved in protein coding but are intergenic enhancer elements (\Cref{fig:ludwig_motifs_8}).

Early hematopoiesis is represented by topics 6 and 8, which are only associated with a handful of genes associated with erythropoiesis (\Cref{table:mello_genes}). HIP1 is known to be crucial to successful hematopoiesis and LYN is involved in the successful proliferation of hematopoetic cells \cite{Oravecz-Wilson2004, OLaughlin-Bunner2001}. Motifs in these regions include ARID3a, known to be expressed in HSCs, and MEIS1, also involved in early hematopoiesis \cite{Zeddies2014,Miller2016, Ratliff2020}. 

These results indicate that BLDA has identified regions which correspond to realistic regions of differential accessibility throughout the process of erythropoiesis. Though the method is not specific, in that many important regions are not represented amongst the results.  



%Later stages of erythopoiesis, characted by ..., are also represented within the dataset. FOXO3, PPP2R1A known to be upregulated towards the end of the process, was identified within topic 3.  Topic 2 contained later genes such as TXNRD2, 




%Regions crucial for erythroid differentiation and survival are represented across the key regions for relevant topics. For example, topic 1 contains the promotor of STAT5a, topic 2 covered the ALAS2 gene, PARK7 is found in topic 3

%Different regions of FOXO3, a gene involved in immune response, were found associated with topics 3, 7, and 1. 

%Topic 2 


\begin{figure}
  \begin{subfigure}[b]{0.5\linewidth}
      \centering
      \includegraphics[width=\textwidth]{plot/ch4/RHAG}
      \slcaption{The RhAG promoter}
      \label{fig:RhAG}
  \end{subfigure}
  \hfill
  \begin{subfigure}[b]{0.5\linewidth}
    \centering
    \includegraphics[width=\textwidth]{plot/ch4/R3HDM4}
    \slcaption{R3HDM4}
    \label{fig:r3hdm4}
  \end{subfigure}

  \slcaption{Accessibility at two keyword regions for topic 2 among cells in the erythropoietic differentiation trajectory. }
    
\end{figure}

\clearpage% Flush earlier floats (otherwise order might not be correct)
%\thispagestyle{empty}% empty page style (?)
\begin{landscape}

  \begin{table}
  %\resizebox{\textwidth}{!}{
    \tiny
\begin{tabularx}{\hsize}{XXXXXXXXX}
  
  \toprule
Category & Topic  1 & Topic  2 & Topic  3 & Topic  4 & Topic  5 & Topic  6 & Topic  7 & Topic  8\\
\midrule
Anti-apoptosis & CTSB & MKL1, BCL2L1, CTSB, NR3C1 & CITED2, PGAP2, TNFAIP3 & PDPK1 & HSPA9, NFKB1 &   &   &  \\
Cellular component involved in apoptosis & ACTN4, KPNB1 & CDH1, KPNB1, GSN & PSMB1, RB1CC1 & ACTN4 & VIM & PSMB1 & DBNL &  \\
Erythroblast enucleation &   & SP1 &   &   &   &   &   &  \\
Erythrocyte differentiation & STAT5A & ALAS2, SLC25A38, BPGM, ERCC2 & CITED2, BPGM &   & HCLS1, NCKAP1L & LYN & BPGM, DYRK3 &  \\
Hematopoiesis \& regulation of cell differentiation & GFI1B, ZBTB16, TXNRD2 & BCL11A, SP1, TXNRD2 & GLRX5, PPP2R1A & TTC7A & IKZF1 & ZBTB16, LYN & IKZF1, CDK6, TTC7A & ZBTB16\\
Heme metabolism & ALAD, BLVRB, TMEM14C & ALAD, ALAS2, BLVRB, FECH, HMBS, SLC25A38, SPTA1, UROD & ALAD & FECH &   &   &   &  \\
Induction of apoptosis & LGALS1, MAPK1, SPN, DAPK2, SAP30BP, VAV1 & KAT2B, RBM38, TP53BP2, AKAP13, ERCC2 & BCLAF1, PPP2R1A, STK17B, TP53BP2, BTG1, NMT1 & CUL3, KAT2B, HIP1 & RBM38, MAP3K5, APP & HIP1 & HIP1 & MAP3K5\\
Iron homeostasis &   & UROD & FBXL5, SLC25A28 &   &   &   &   &  \\
Oxygen homeostasis/ response to hypoxia or to oxidative stress & ACTN4, MLH1 & HMBS, STAT5B, ERCC2, NRF1, PTK2B & CAT, CITED2, MLH1, OXSR1, PARK7 & ACTN4, PTK2B & ECE1, NFKB1, NR4A2 & IPCEF1 &   & CAT, IPCEF1\\
Primitive hematopoiesis & STK3 &   &   &   &   &   &   &  \\
\bottomrule
  \end{tabularx}
  %}

  \slcaption{Genes from \textcite{Mello2019} represented in the closest genes set of 500 keyword regions for each of the eight BLDA topics grouped by function.}
  \label{table:mello_genes}
\end{table}
\end{landscape}
\clearpage% Flush page


%  \begin{figure}
%  \centering
%  \includegraphics[width=\textwidth]{plot/ch4/GATA3.pdf}
%  \caption[GATA3 Promotor]{Coverage plot of the GATA3 promoter. BLDA identifies this region as discremintory for CD4 and CD8 T cells, and coverage plot indicates that it is differentially accessible between erythopoetic and terminally differentiated lymphocytes.}
%  \todo{Why is the other beds peaking through???}
%  \label{fig:gata3}
%\end{figure}


%Topic 1:
 % - Stat5a promotor is one of the most important regions. Stat5a is 
 %   -  regulation of RBC production by expansion of earlier progenitor and stem cell populations.
 %   - topic 1 esxpression starts early and ramps up through erythropoiesis
 %   - (and 2) share a peak with MiR584. What role does this play?


%  Topic 8 (and 5)
%    - CGAS is widely expressed but especially in hematopoetic progeneitors (https://www.frontiersin.org/articles/10.3389/fimmu.2020.573915/full)

%Topic 2
%  - Many of the systems like UROS, CCND3, VEGFA, and TMCC2 are not found in this system.
%  - RHAG is found, peak accessibility in BasoE and the topic it is found in peaks with BaseO. 
%    - Expressed in mature 
%  - HDAC6, some role in enucleation,not completelysure (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5451330/)
%  - R3HDM4 is accessible in all of the cells, but differentially so. 
  %  - It's expession (RNA-seq) is associated with erythropoiesis https://genevatool.org/gse_description?gene_name=R3HDM4&gse_id=GSE102182
  %  - Experssed in mature Erythrocytes https://genevatool.org/gse_description?gene_name=R3HDM4&gse_id=GSE63703



% Topic5:
 % - Has a promotor for GATA3. Maybe because this is not accessible in CD8 and CD4? But in everything else? 
   % - Motifs found in almsot every other cell type
 % - Actually also important for t cells https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3688666/

%Early: 
%  - Arid3a / bright (6)
%  - MEIS1 is expressed prefernetially in HSCs (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3525022/)

\section{Discussion}

Recently, the use of \gls{atac} to characterize chromatin accessibility in varied cell systems has resulted in an excess of high quality data. However, it is currently difficult to integrate these data and perform inference for common regulatory programs. In this chapter, I adapted the cisTopic approach for using \gls{lda} to simultaneously infer weighted collections of co-accessible regions and the cell types in which they are active. I show that the method, BLDA, is able to identify a similar number of differentially expressed regions (as defined by EdgeR) as the established \gls{scatac} pipeline in a collection of pseudobulked single cell experiments with known cell types. The modification was essential to its comparative success, as a naive implementation of the cisTopic algorithm using region thresholding and one-hot encoding did not identify any of the same differentially accessible regions as the single cell analysis. BLDA additionally identified topics which were much more specific to target cell types than the naive implementation, as is expected in a purpose-built dataset of dissimilar cell types (\Cref{fig:pb_no_thresh_lot_topics}). This simple dataset shows that BLDA is able to recreate some of the success of the single cell method in bulk samples, so I next turn to a well understood system of differentiating cell types. Erythropoiesis, the process by which \glspl{hsc} differentiate into red blood cells, or here more specifically immature erythroblasts which have not yet undergone enucleation, has an extensive literature documenting genes which are up and down regulated at various check points \cite{Ludwig2019}. We take sequencing data from \textcite{Ludwig2019} and \textcite{Corces2016} and create a pseudo-timed differentiation trajectory incorporating hematopoietic precursors and erythropoietic check points, as well as stages of the lymphoid differentiation pathway incorporating CD4 and CD8 positive T cells. I show that in this system as well, the BLDA method out performs a naive implementation of cisTopic, identifying specific topic loadings that are enriched for the regulatory biology of erythropoiesis. Inferred topics identify regions of the genome which are shown to be differentially accessible in similar patterns to topic loadings across pseudo-time such as \Cref{fig:RhAG} and \Cref{fig:r3hdm4}. 

Having established that BLDA has power to find biologically enriched pathways within topic loadings, the next chapter applies the algorithm to large scale datasets of ATAC-seq experiments like ENCODE and uses them to investigate cell types with relatively unknown regulatory grammar.  

\section{Data and Code Availability}

Single cell sequencing experiments from \cite{Buenrostro2015} are available via SRA, and scripts for performing peak calling and pseudobulking are distributed with the BLDA package at \url{https://github.com/Chris1221/BLDA}. This package also contains functionality for creating count matrices from BAM files and bigWig tracks in order to run the adapted analysis pipeline. A modified version of cisTopic necessary for running these analyses is available at \url{https://github.com/Chris1221/cisTopic_bulk}. 

Data for the erythropoiesis developmental trajectory is available from GEO, with more details about specific file formats available from the original publications \textcite{Corces2016} and \textcite{Ludwig2019}.

An implementation of LanceOTron peak caller for batch data is available from \url{https://github.com/Chris1221/LanceOTron}.

A Snakemake pipeline for replicating the entirety of the analyses presented in this chapter is available in the chapter\_3 subdirectory of \url{https://github.com/Chris1221/thesis_lda} repository.

\section{Acknowledgments}

The original conception of the BLDA extension, that is incorporating read counts rather than binary accessibility into the count matrix, came from Alastair Smith. Single cell experiments were downloaded and combined into a single BAM file by Emine Ravza Gur. Alignment and quality control of the datasets from \textcite{Corces2016} and \textcite{Ludwig2019} was performed by Damien Downes. 