%\begin{savequote}[8cm]
%\textlatin{Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...}

%There is no one who loves pain itself, who seeks after it and wants to have it, simply because it is pain...
%  \qauthor{--- Cicero's \textit{de Finibus Bonorum et Malorum}}
%\end{savequote}

\chapter{\label{ch:1-aaa} Ancient Admixture into Africa from the Ancestors of non-Africans} 

\minitoc




\begin{quote}
    \noindent \small Genetic diversity across human populations has been shaped by demographic history, making it possible to infer past demographic events from extant genomes. However, demographic inference in the ancient past is difficult, particularly around the out-of-Africa event in the Late Middle Paleolithic, a period of profound importance to our species' history.  Here we present {\tt SMCSMC}, a Bayesian method for inference of time-varying population sizes and directional migration rates under the coalescent-with-recombination model, to study ancient demographic events. We find evidence for substantial migration from the ancestors of present-day Eurasians into African groups between 40 and 70 thousand years ago, predating the divergence of Eastern and Western Eurasian lineages.  This event accounts for previously unexplained genetic diversity in African populations, and supports the existence of novel population substructure in the Late Middle Paleolithic. Our results indicate that our species' demographic history around the out-of-Africa event is more complex than previously appreciated.
\end{quote}

\newpage

\printglossary

\section{Introduction}

Methods to characterize demographic history from genetic data alone form a useful and independent complement to archaeological approaches \cite{Nielsen2017a}, and many methods have been developed for this purpose \cite{Li2011,Pickrell2012,Rasmussen2014,Schiffels2014,Mathieson2014,Steinrucken2015,Malaspinas2016,Chikhi2018,Speidel2019,Kelleher2019,Wang2019a,Albers2019}. 
A particularly interesting time in human history is the end of the Middle Paleolithic approximately $\sim$60 \gls{kya}, which saw the divergence of the most deeply sampled lineages of human genetic variation, introgression from multiple archaic sources, and the expansion of anatomically modern humans \gls{ooa}. 
As archaeological evidence and ancient DNA from this period are scarce, inference of demography from present-day genetic data is potentially very informative, though technically challenging. Here, we use the previously developed approach, \gls{smc2}, to infer population size and directional migration in a unique and dynamic period of ancient human development.

This section is structured as follows. Firstly, I give an introduction to pertinent historical and anthropological theories relating to this period of time so as to orient the reader. Secondly, I introduce competing approaches for the inference of ancestral recombination graphs and motivate the usage of \gls{smc2} for this application. Following this, I outline my contributions to this area of research, which involve the identification and characterization of a putative directional migration from the ancestors of modern day Eurasians to the ancestors of modern day Africans. 


\subsection{Out of Africa and the peopling of Eurasia}

An abundance of archaeological and genetic evidence has shown that the continent of Africa is the historical source of all modern humans \cite{Lopez2015}. The continent contains within it more genetic sequence diversity than any other region of the world, so much so that the two haplotypes of a single African genome are less similar than any two haplotypes outside of the continent \cite{Mallick2016}. The population structure in the continent has its roots in the transition from the middle to late Pleistocene, with hunter gatherers groups from Southern Africa including the Khomani San and Ju'|Hoan diverging in large part from the remainder of modern African populations between 200 and 250 \gls{kya} \cite{Lipson2019}. Additionally, recent evidence suggests that deeply diverged archaic lineages have substantively contributed to at least some modern Western African populations \cite{Durvasula2019}. Populations on the content were, therefore, highly structured and geographically localised many tens of thousands of years before we have evidence for the first \gls{amh} leaving the continent.    

Evidence from climate science suggests that a combination of a gradual shift away from aridity in Northern Africa as well as short term dry-wet cycles may have motivated both global and local range expansions \cite{Schaebitz2021, Timmermann2016}. The most pertinent of these range expansions is the migration \gls{ooa} and into Eurasia, an event which will form the basis of modern population structure around the globe. These migrants were probably diverged from sister populations within the continent for many tens of thousands of years before their eventual dispersal into the Levant and beyond \cite{Bergstrom2019, Schiffels2014}. Though these individuals were by far the most successful group of migrants, some fossil evidence exists placing \gls{amh} in regions of the globe such as Fuyan cave in China between 80 and 120 \gls{kya}, between 63 and 73 \gls{kya} in Sumatra, and 65\gls{kya} in Australia, all of which would be earlier than explainable with this single migration; however, genetic analysis of diverse populations has yet to identify any solid evidence for these earlier migrations contributing to modern diversity \cite{Skoglund2018}.

The population which formed this successful range expansion out of Africa experienced at least one, and potentially multiple, breeding events with Neanderthals \cite{Sankararaman2012}. Eventually, the group split into two distinct subpopulations around the time of the Ust'-Ishim individual, approximately 45 \gls{kya} \cite{Fu2014}. One of these populations went East, forming the basis of for East Asians and Aboriginal Australians, while the other went West, forming the initial Upper Paleolithic European hunter gatherers \cite{Skoglund2017,Lipson2017}. These were not the only derivative groups from the original \gls{ooa}, as another earlier diverged population popularly known as ``Basal Eurasians'' are thought to have branched before the initial contact with Neanderthal populations and contribute to later European population structure \cite{Lazaridis2014}. From this point on, diversification occurred on a highly regional basis, beyond the scope of this brief review. 

Many questions about this process of diversification remain unanswered, however. 

% Remaining questions about how these groups interacted...? What about other theories of the back migration?

\subsection{The Ancestral Recombination Graph (ARG) and admixture inference in the ancient past}

The phylogenetic trees over a set of samples as they change along the genome through recombination, collectively referred to as the ancestral recombination graph (ARG) \cite{Griffiths1997a,Rasmussen2014}, record all information about the samples' evolutionary history.  This history itself is shaped by the population's demography, a statistical relationship that is quantified by the coalescent-with-recombination (CwR) model \cite{Griffiths1997a}.  The ARG is a complex data structure which is only weakly constrained by the observed genetic polymorphisms, making inference of demography difficult. By making approximations to the CwR, for instance by making an independent-sites assumption, efficient parametric inference of demography becomes possible \cite{Excoffier2013,McVean2005}.  Methods including {\tt PSMC} \cite{Li2011}, {\tt diCal} \cite{Steinrucken2015} and {\tt SMC++} \cite{Terhorst2015} allow non-parametric inference of demography under a closer approximation to the CwR, but one that does not include gene flow between populations. {\tt MSMC} \cite{Schiffels2014} introduced the cross-coalescent rate and {\tt MSMC-IM} described how to interpret this rate in the context of a isolation-migration model to estimate a migration rate between populations\cite{Wang2019a}. However, these methods are not well suited for estimating directional migration rates. 

Here we extend {\tt SMCSMC} (Sequential Monte Carlo inference of the Sequentially Markovian Coalescent, \cite{Henderson2018}) to allow inference of directional migration.  {\tt SMCSMC} is a Bayesian method that uses a particle filter to explicitly sample from the posterior distribution of ARGs over multiple diploid samples under the full CwR model.
Since particle filters operate by simulating latent variables (here the ARG) under the statistical model of interest, it becomes possible to handle complex demographic scenarios.  We exploit this by extending the CwR model to include time-varying directional migration rates in a two-island demographic model.  We use the posterior sample of ARGs including migration events to update the parameters of the demographic model, using either expectation-maximization or a variational Bayes procedure, and iterate these steps until convergence.   We apply {\tt SMCSMC} to estimate directional migration rates in whole genome sequencing data from the Simons Genome Diversity Panel (SGDP) \cite{Mallick2016} and the Human Genome Diversity Panel (HGDP) \cite{Bergstrom2019} to investigate population structure around the OoA event.

\section{Results}

\paragraph{Substantial Migration from Eurasian to African Ancestors} We use {\tt SMCSMC} to analyse pairs of individuals from the SGDP and simultaneously infer migration rates and effective population sizes ($N_e$) under a two-island model with directional migration.  Population sizes and migration rates are modeled as piece-wise constant across 32 exponentially spaced epochs from 133 to 133016 generations in the past, corresponding to 3.8 thousand to 3.8 million years ago (3.8kya--3.8Mya) using a generation time $g=29$ years \cite{Fenner2005}.  We find that the method infers high rates of migration from descendants of the OoA event ('non-Africans') to Africans, but not in the opposite direction, in the period $30$--$70$kya corresponding to the Late Middle Paleolithic (Fig.\ \ref{migrationplot}). In populations from the Niger-Kordofanian and Nilo-Saharan language groups, comprising the majority of the population on the African continent, the peak inferred migration rate from Eurasian populations ($2.5$--$3.0\times 10^{-4}$ and $3.5$--$4.0\times 10^{-4}$, in units of proportion of the target (ancestral African) population replaced per generation) most frequently falls in the epochs spanning 35--45kya, while peak migration rates in the opposite direction are substantially lower ($0.5$-$1.0\times 10^{-4}$) and occur earlier, in the epochs spanning $55$--$70$kya (Supplemental Fig.\ \ref{fig:peaks}). Populations in the Afroasiatic language group show evidence of large amounts of directional migration in the Holocene (Supplemental Fig.\ \ref{sgdp_mig}), which is consistent with previous findings of relatively recent European introgression into these populations \cite{Busby2016, Fan2019}. 

To assess the impact of errors introduced by statistical phasing, which was used to phase the SGDP data, we repeated the analyses above on a subset of physically phased individuals from the Human Genome Diversity Project (HGDP) \cite{Mallick2016} (Supplemental Section \ref{hgdp_section}). This data set comprises individuals from four African (Yoruban, San, Mbuti, and Biaka) and nine non-African populations (Druze, Han, Karitiana, two Papuan populations, Pathan, Pima, Sardinian, and Yakut). {\tt SMCSMC} results in the HGDP are qualitatively similar to those in the SGDP (Fig.\ \ref{migrationplot}b, Supplemental Fig.\ \ref{fig:peaks}a). Inferred migration rates are, in general, lower in HGDP data than when using matched SGDP samples (Figs.\ \ref{migrationplot}a,b and \ref{fig:both}, demographic inference in matched sampled in Supplemental Fig./ \ref{fig:hgdp_sgdp}), but in all cases, the migration rates from Eurasia to Africa are substantially higher than in the opposite direction, consistent with the findings in the SGDP (Supplemental Fig.\ \ref{fig:both}). 

We asked whether {\tt SMCSMC} has power to detect a large back-migration event in the Late Middle Paleolithic and distinguish it from other demographic scenarios. To answer this we used {\tt SCRM} \cite{Staab2015} to simulate a gigabase of sequence data under a two-island demographic model,
%with a split time of 385kya,
with effective population sizes chosen to be comparable to typical African and Eurasian populations as inferred from real data. 
%We simulated an early split time to allow {\tt SMCSMC} to resolve the split times of the populations without undue bias. \textcolor{red}{\bf This explanation sounds weak.  It suggests that we're making simulating an easier case than reality, throwing doubt on our simulations.  Thoughts?} 
To this we added a $10$ky pulse of forward, backward or bidirectional migration of varying strengths, with the midpoint of the migration pulse within the range $40$ to $70$kya.  To quantify the inferred amount of migration we calculate the integrated migration fraction (IMF), defined as one minus the probability that a lineage in the destination (e.g.\ African) population traced backwards in time remains in that population across a given epoch according to the migration model (see Methods).  For the simulations, we chose the most recent 100kya as epoch, and used scenarios with IMFs ranging from $0$ to $0.593$. For each simulation we report the inferred IMF in both the forward and backward direction (Fig.\ \ref{fig:sim}); full results are given in Supplemental Section \ref{simproc}.  We find that {\tt SMCSMC} has good power to detect backward migration pulses up to $60$kya (median ratio of inferred and true IMF, $0.91$), while power drops off at $70$kya (IMF ratio $0.46$). In the pure backward migration case, some forward migration is falsely inferred, but this is always substantially less than the inferred backward migration (median ratio inferred forward to true backward IMF, $0.37$; true migration peak $\leq 60$kya).  However, in the case of true forward migration as well as bidirectional migration, roughly equal mixtures of forward and backward migration are inferred (Fig.\ \ref{fig:sim}). We conclude that in the epoch $40$--$70$kya the forward and bidirectional scenarios are difficult to distinguish from each other, but both can be distinguished from backward migration, the only scenario resulting in substantially different inferred backward and forward migration.

To validate the existence of the migration pulse, though not its direction, we next analyzed the same data using MSMC, which is widely used to estimate gene flow in the ancient past by estimating the relative cross-coalescent rate (RCCR) between two populations \cite{Schiffels2014,Fan2019, Pagani2015, Raghavan2015}. We use the updated implementation MSMC2 recommended by the authors and first published in \cite{Malaspinas2016}. Each of the {\tt SMCSMC} analyses are repeated using MSMC2 to estimate effective population size and RCCR (Supplemental Figs.\ \ref{sgdp_mig}, \ref{fig:both}, \ref{sgdp_ne}). Consistent with previous analyses conducted with MSMC2, our estimates show high RCCR in the Late Middle Pleistocene in both the SGDP and the HGDP (Fig.\ \ref{migrationplot}c,d) \cite{Fan2019, Bergstrom2019}. These observations confirm the existence of a substantial pulse of ancient gene flow between Eurasians (Han Chinese) and Africans.

\paragraph{Migration Pre-dates East-West Eurasian Divergence}

To assess whether the inferred back-migration shows variation across the descendants of the OoA event, we repeated the analyses using three representative non-African groups in the SGDP: Han Chinese, French European, and Papuans.  Since simulations show that {\tt SMCSMC} has little power to detect migration predating 70kya, and to exclude Holocene migration, the epoch we use to calculate real-data IMFs comprise the period of peak inferred migration up to the period of diminishing power (30--70kya); we use this epoch for all subsequent analyses. Inferred IMFs are not significantly different between Han Chinese and European populations in non-Afroasiatic populations (p=0.14, two-tailed paired t-test; Figs.\ \ref{migrationplot}h and \ref{sgdp_heatmap}, Table \ref{average_sgdp_migration_table}), consistent with migration occurring before the European-East Asian split approximately 40kya \cite{Mathieson2014}.  The contribution of this admixture event to extant African genetic variation is substantial; the estimated IMFs indicate that for individuals in the major African language groups, approximately a third of ancestral lineages trace their ancestry through the proto-Eurasian population (Niger-Kordofian group, $0.35\pm 0.04$; Nilo-Saharan groups, $0.41\pm 0.03$; Table \ref{average_sgdp_migration_table}). When we estimate these proportions using a Papuan sample to represent non-African descendants we find slightly but significantly smaller values compared to estimates using either the Han Chinese or European populations (mean difference of $0.029 \pm 0.002$, p=9.2$\times$10$^{-15}$, and $0.025 \pm 0.004$, p=$2.3\times 10^{-10}$, paired t-tests, Supplemental Table \ref{average_sgdp_migration_table}, \ref{sgdp:papuan_imf}). Similarly, in the HGDP, inferred migration in both Papuan groups (Sepik and Highlands) was 0.025 $\pm$ 0.004 (p=$1.4\times10^{-6}$) lower than French and Han (Supplemental Table \ref{hgdp:papuan_imf}).  We comment on this observation in the Discussion. 

\paragraph{Directional Migration Explains Excess Inferred African Genetic Diversity 100kya} Previous studies looking at effective population sizes ($N_e$) in human ancestral populations have consistently reported inflated inferences in African populations approximately 100kya, often hypothesized to be due to unaccounted-for population substructure within Africa \cite{Li2011,Schiffels2014}. We use {\tt SMCSMC} to analyze African individuals paired with an individual from one of three non-African populations (Han Chinese, French European, and Papuans) and infer $N_e$ for the African ancestral population under a two-island model with directional migration.  Each analysis was repeated three times to assess the contribution of stochastic sampling to the inferences (Figs.\ \ref{neplot}, \ref{sgdp_ne}, per population $N_e$ in Supplemental Fig. \ref{fig:individual_pop_sizes}). {\tt SMCSMC} infers substantially lower African $N_e$ than {\tt MSMC} in the period $80$kya--$300$kya.  In addition, while {\tt MSMC} inferences show convergence of African and Eurasian ancestral $N_e$ estimates only around $300$kya, inferences from {\tt SMCSMC} indicate convergence at $150$kya (Fig.\ \ref{neplot}a), closer to the hypothesized time of the diversification of the ancestral lineages prior to the main out-of-Africa migration episode \cite{Timmermann2016, Malaspinas2016}. The same analysis on physically phased samples from HGDP show that these results are not driven by errors due to statistical phasing (Fig.\ \ref{fig:both} and Supplemental Section \ref{hgdp_section}). When we used {\tt SMCSMC} to infer both African and European $N_e$ under a single-population model without migration, $N_e$ estimates were comparable to those from {\tt MSMC} (Fig.\ \ref{neplot}b), indicating that the {\tt SMCSMC} inferences are not driven by methodological biases particular to {\tt SMCSMC}.

To more directly support the interpretation that the lower African $N_e$ inferred by {\tt SMCSMC} is due to appropriate modeling of directional migration, we again used coalescent simulation with {\tt SCRM} to investigate various migration scenarios and their effects on inferred African $N_e$. Using the simulation framework as above, we examine $N_e$ estimates inferred under a two-island model with migration, and in addition $N_e$ separately inferred for each of the two simulated populations under a single-population model (Supplemental Section \ref{simproc}).  Focusing on single-population inferences, we found that for simulated African populations that had received substantial migration from the simulated Eurasian population either through backward or bidirectional migration, inferred $N_e$ values indeed were substantially inflated compared to true values (Fig.\ \ref{neplot}c,d), while this effect was not seen when forward (African-to-Eurasian) migration was simulated (Fig.\ \ref{neplot}e).  
Similarly, single-population Eurasian $N_e$ estimates were inflated in the presence of forward and bidirectional migration, but not backward migration (Supplemental Figs.\ \ref{fig:backsim}--\ref{fig:fwdsim}).
In contrast, when using a model that includes migration, inferred African $N_e$ do not show inflation in any of the three scenarios (Fig.\ \ref{neplot}c-e). We conclude that the inferences from {\tt SMCSMC} and {\tt MSMC} are compatible with substantial back-migration from ancestral Eurasians into Africans, but not substantial bidirectional or forward migration.

\paragraph{Less Gene Flow to Central and South African Hunter-Gatherers} We infer substantial Eurasian back-migration into all African groups, however the inferred IMFs for individuals from Khoe-San populations are significantly lower than for any other group (difference with Niger-Kordofians, $0.14 \pm 0.02$, $p = 4.4 \times 10^{-14}$; difference with Nilo-Saharans, $0.20 \pm 0.03$, $p = 6.9 \times 10^{-9}$, two-tailed t-test, Table \ref{table:sgdp_pairwise}). To further support this observation we used {\tt MSMC} to estimate the relative cross-coalescent rate (RCCR) for several populations, and find evidence for gene flow between Yorubans and Eurasians that is not shared with the Khoe-San individuals in either the SGPD and the HGDP (Fig.\ \ref{migrationplot}c,d). These results are consistent across Eurasian donor populations (Fig.\ \ref{fig:hgdp_sgdp}). The Khoe-San individuals are particular outliers, whose ancestors are inferred to have experienced approximately half the amount of admixture seen in Nilo-Saharan and Niger-Kordofanian groups (Fig.\ \ref{sgdp_heatmap}). 
In addition, we find that the Mbuti and Biaka, both Central African hunter-gatherer populations, show levels of Eurasian gene flow that are intermediate between levels observed in the Khoe-San and Yorubans (Fig.\ \ref{migrationplot}a,b, Supplemental Table \ref{average_sgdp_migration_table}).  This is mirrored by inferred IMFs for Central African Hunter Gatherers, which are significantly lower than other Niger-Kordofanian groups (difference $-0.08 \pm 0.03$, $p = 1.2 \times 10^{-3}$, Table \ref{table:sgdp_pairwise}), possibly reflecting the proposed early split times of the Mbuti and Biaka from the remainder of ancestral African populations between 60 and 200kya \cite{Patin2017, Lipson2019}. 

\paragraph{No Evidence for Excess Neanderthal Ancestry} Previous studies have proposed that a backflow from Eurasia may have brought Neanderthal ancestry into African populations \cite{Chen2020}. To assess whether the proposed Late Middle Paleolithic back migration might have introduced Neanderthal material, we analyzed a Yoruban and a French individual using {\tt SMCSMC} to draw a sample from the posterior distribution of ARGs, isolated the marginal trees containing an inferred back-migration event in the epoch $30$--$70$kya, and reported the inferred admixture tracts (``segments'', Supplementary Section \ref{dstats_section}). To assess whether the identified segments are plausible, we confirmed that their length distribution is consistent with IMF and timing of the migration inferred by {\tt SMCSMC} (Supplemental Section \ref{dstats_section}.1, \ref{fig:length}), and, as expected, we found that these African segments with putative Eurasian ancestry tend to be more closely related to a Eurasian sample than another representative of the same African population (Table \ref{dstats:a1}, Supplemental Fig. \ref{fig:f3}, \ref{fig:f3_admix}) in a global dataset of modern and ancient individuals compiled by the Reich group (see URLs). Within these African segments that are likely enriched for material with Eurasian ancestry, we then used $D$ statistics \cite{Patterson2012} to identify enrichment for Neanderthal material compared to an African background. We find no evidence for gene flow with a Vindija Neanderthal on the Mbuti baseline, or when compared to a different Yoruban (Table \ref{dstats:a5}, \ref{dstats:a4}). We additionally find no evidence for increased affinity to the Vindija Neanderthal when compared to the Altai, as would be expected if the material were descended from admixing Eurasians (Table \ref{dstats:a6}). However, we find that restricted to the identified segments, $D$ statistics have power to detect evidence for the known admixture from Vindija into a French individual (Fig.\ \ref{dstats}), suggesting that lack of power does not explain the lack of evidence we find for Neanderthal admixture into Africans.  In addition, we find no differences in affinity to Neanderthals or Denisovans between the variants which fall in segments and the whole genome (Fig.\ \ref{dstats}d). Taken together, this suggests that Eurasian-derived segments of the African genomes are not enriched with Neanderthal material.

\section{Discussion}
We have developed an approach for estimating demographic parameters and ARGs from whole genome sequence data, which can handle inference in complex demographic models, and implemented this in the software program {\tt SMCSMC} \cite{Henderson2018}. We used {\tt SMCSMC} to investigate ancient migration rates and population substructure, and found evidence for a substantial admixture from ancestors of present-day Eurasian populations into African populations in the Late Middle Paleolithic.

Our analysis suggests that a population ancestral to present-day Eurasians contributed as much as a third of the genetic material in many modern African populations. We find no difference in inferred admixture proportions when using French Europeans or Han Chinese as extant representatives of the donor population, indicating that the admixing population must have split from the out-of-Africa population before the East/West Eurasian divergence, implying a lower bound on the timing of the admixture of approximately 40kya \cite{Mathieson2014}. It appears that our results suggest that the migrating population was more similar to present-day French and Chinese populations than to Papuans.  However, up to 5\% of the genomes of some present-day Papuans have been suggested to derive from archaic introgressions \cite{Sankararaman2016}, and these contributions will have reduced the inferred levels of admixture into Africans when using Papuans as a representative of the Eurasian ancestors.
The alternative explanation, of an earlier divergence of Papuans and Eurasian ancestors, is possible but contested; in light of documented Eurasian admixture into Oceania, the effects of this early isolation are likely to be small relative to the large confounding effects of Denisovan admixture \cite{Malaspinas2016, Nielsen2017a}.

The proposed period of admixture has biased previous inferences of the African population sizes. We show that including directional migration into the model resolves previously unexplained high inferred $N_e$ in the period $80$ to $300$kya. It is well known that effective population size estimates are biased in the presence of population substructure and migration \cite{Chikhi2018, Li2011}. We use simulations to show that the proposed admixture event indeed causes an increase in estimated $N_e$ in analyses that do not explicitly model migration.  Correctly modeling of directional migration recovers the correct $N_e$, and allows us to infer a more recent split time between the two populations than indicated by previous analyses, although we did not attempt to formally estimate this time of divergence.

We found that not all populations in Africa have been equally affected by the proposed migration event. While the ancestors of Niger-Kordofanian and Nilo-Saharan populations show evidence of similar levels of Eurasian admixture, the ancestors of Central African and South African hunter-gatherer populations show markedly lower levels.  The date of genetic diversification of both the Central Hunter Gatherers and Khoe-San (SAHG) is contested \cite{Lipson2019}, but a date of $100$kya has been proposed \cite{Schlebusch2012}, providing a putative upper bound on the main admixture event.  Our simulations indicate that {\tt SMCSMC} has little power to detect the impact of migration events occurring more than $70$kya, providing an additional upper bound on the time of the migration episode, or the fraction of it that left a sufficiently distinct imprint on extant genetic material.

Compared to the remainder of the Niger-Kordofanians and Nilo-Saharans on the one hand, and the SAHG populations on the other, the Mbuti and Biaka show intermediate levels of admixture. Of these populations, the Biaka show slightly higher levels of admixture than the Mbuti, which is likely due to the well-documented admixture from Western African groups not shared with the Mbuti \cite{Batini2011}. The lower levels of admixture in Mbuti and Biaka compared to Niger-Kordofian and Nilo-Saharan populations imply at least partial diversification of the former at the time of the migration, placing an upper bound on the timing. However, dating the diversification of these groups is difficult. Recent estimates using $f$ statistics place the split concurrent with the San in a large-scale early expansion 200-250kya \cite{Lipson2019}, while older data consistently report an earlier split time between 50 and 90 kya \cite{Patin2018}. Further clarity on the early structure and diversification of hunter-gatherer populations are necessary to interpret their interactions with Eurasian migrants. The  Afroasiatic populations on the other hand show high levels of admixture, which also appears to be of much more recent origin, and it appears likely that this is the result of extensive admixture from Eurasian populations during the Holocene \cite{Busby2016, Fan2019}. 

It has previously been suggested that Eurasian back-migration may be responsible for Neanderthal material in Africans \cite{Chen2020}; however, we find no evidence for enrichment of Neanderthal-like material in putatively Eurasian-derived genomic segments in Africans, indicating that Neaderthal introgression into Eurasians occurred after the African introgression event we study here, or that further population structure in the Eurasian ancestral population precluded substantial transmission of Neanderthal material into Africa.

Our findings are consistent with several other published observations. Migration rate estimates using {\tt MSMC-IM} revealed high levels of admixture at times comparable to our results \cite{Wang2019a}. The coalescent intensity function additionally shows similar histories between sub-Saharan African and Eurasian groups with high coalescent intensity in epochs consistent with our inference and those of {\tt MSMC-IM}, supporting both an early split between the groups and a substantial replacement of genetic material more recently than $\sim100$kya \cite{Albers2019}. Evidence has been mounting for multiple migrations into the Eurasian continent, possibly mediated by climatic drivers \cite{Timmermann2016, Pagani2016}. Eurasian backflow during the Holocene has been well established \cite{Lopez2015, GallegoLlorente2015}, but earlier migrations have also been proposed before based on observations of the spatial distribution of Y chromosome and mitochondrial haplogroups \cite{Altheide1997, Hammer1998, Cruciani2002, Chandrasekar2007, Cabrera2018, Hervella2016, Haber2019}. At the same time, evidence has been mounting for extreme heterogeneity in the history of sub-Saharan Africans, with several unsampled population theorised to have contributed at various points in the past \cite{Lipson2019, Durvasula2019, Speidel2019}. In light of these recent studies, the observations in this paper add to a growing body of evidence for complex population structure and migration surrounding the Out of Africa event leading to a substantial replacement of the African population in the Late Middle Paleolithic.  


\section{Methods}


\paragraph{A Particle Filter for Demographic Inference} Details of the Sequential Monte Carlo for the Sequentially Markovian Coalescent ({\tt SMCSMC}) algorithm have been previously published \cite{Henderson2018} (see the URLs for an implementation). Briefly, {\tt SMCSMC} builds an approximation of the posterior distribution of genealogical trees along the genome using a particle filter, a method also known as sequential Monte Carlo. It does so by simulating a number of sequences of genealogical trees (particles) under a fixed set of demographic parameters $\theta$, using the sequential coalescent sampler {\tt SCRM} \cite{Staab2015}. Simulated recombination events may change the local trees along the sequence. Particles are then weighted according to their conditional likelihood given observed polymorphisms.  To avoid sample depletion, the set of particles is regularly resampled, which tends to remove and duplicate particles with low and high weight respectively.  To further increase the efficiency of the procedure, the resampling procedure targets not the partial posterior distribution that includes polymorhpisms up to the current location, but also includes a "lookahead likelihood" term that approximates a particle's likelihood's dependence on subsequent polymorphisms, while ensuring that the estimate of the posterior tree distribution remains asymptotically exact.  From a sample of trees from the posterior distribution, Variational Bayes (VB) or Stochastic Expectation Maximization (SEM) is used to update the estimates of demographic parameters $\theta$. This is repeated over a given number of iterations, or until $\theta$ have converged.

To add the ability to infer time-varying migration rates, we exploit the capabilities of {\tt SCRM} to simulate ARGs under complex demographic scenarios, and collect sufficient statistics (migration opportunity, and number, time and direction of simulated migration events) for each particle.

We use {\tt SMCSMC} to infer effective population sizes and migration matrices in pairs of unrelated individuals from the phased release of the Simons Global Diversity Panel. We set a uniform recombination rate of $3\times10^{-9}$ and a neutral mutation rate of $1.25\times10^{-8}$, both in units of events per nucleotide per generation; previous results indicate that modeling recombination
hotspots minimally affects results \cite{Li2011}. To reduce the number of iterations to convergence, we initialise the particle filter with an approximation of human demographic history (Supplemental Fig. \ref{smc2demog}).
We seed the model with an initial constant symmetric  migration rate of 0.0092 ($M_{i,j}$; proportion per generation of the sink population replaced by migrants from the source backwards in time). We arrive at this value through simulation (Supplemental Section \ref{simproc}, Supplemental Figs. \ref{fig:intsim}, \ref{init_yri}).

\paragraph{Multiply Sequential Markovian Coalescent} We use {\tt MSMC2} to estimate the effective population size of pairs of African and Eurasian individuals using default configurations and scripts provided in {\tt msmc-tools} (see URLs) \cite{Schiffels2014, Wang2019a}. We use a fixed recombination rate in line with our {\tt SMCSMC} analysis and skip ambiguously phased sites. Twenty iterations are performed by default. We additionally compute the relative cross-coalescent rate to examine relative gene flow by transforming the coalescent rates generated by {\tt MSMC2} as indicated in the software documentation.



\subsection{Inferring population size and migration rates in the Simons Genome Diversity Panel}

This section describes analysis of the \gls{sgdp} with both {\tt SMCSMC} and MSMC. {\tt SMCSMC} version 1.0.1 was installed from the conda package manager (also found at \url{https://github.com/luntergroup/smcsmc/releases/tag/v1.0.2}), {\tt MSMC2} version 2.1.2 was installed from Github (found at \url{https://github.com/stschiff/msmc2/releases/tag/v2.1.2}) and all analyses were performed on the Oxford Biomedical Research Computation cluster.

We download prephased sequencing data from {\tt https://sharehost.hms.harvard.edu/genetics/reich_lab/sgdp/phased_data/} and mask for the strict accessibility mask from the 1000 genomes project. We additionally mask for any sites absent Chimpanzee ancestry due to a known issue with the phasing algorithm \cite{Wang2019a}. We do this masking in {\tt vcftools}. We use \gls{smc2} to convert the sequence data from VCF to seg file format, a format very similar to \gls{MSMC} format. We provide a script to convert from seg file format to \gls{MSMC} file format as well. Unless otherwise noted, the names of individuals used in this paper are the first in their population (i.e. an individual named Yoruban is {\tt S\_Yoruba-1} in the SGDP nomenclature, full list in \ref{samples}).  We select two diploid individuals from each population in Africa  and infer piecewise constant population size and directional migration rates. Specifically, we use the following options for \gls{smc2}:

\begin{verbatim}
smc2 -c -chunks 100 -no_infer_recomb -nsam 4 -I 2 2 2 -mu 1.25e-8 -rho 3e-9 \
    -calibrate_lag 1.0 -EM {EM} -tmax 3.5 -alpha 0.0 -apf 2 -N0 14312 -Np {Np} -VB \
    ${DEMOGRAPHIC_MODEL} -P 133 133016 31*1  -arg -o ${OUTPUT} -segs ${SEGS}
\end{verbatim}

In order, we invoke the use of a QSUB cluster with {\tt -c} and split our analysis into 100 chunk. We do not infer recombination sites along with the demographic model in order to reduce runtime. Four haploid samples, two from each population, are analysed with a mutation rate of 1.25 $\times 10^{-8}$, a recombination rate of 3 $\times 10^{-9}$, and accumulating events for one unit of survival time along the sequence. We use a given number of epochs for parameter units, and bound the upper limits of the trees at 3.5 times the effective population size (set to 14312). We use the look-ahead likelihood to guide the resampling process for a given number of particles {\tt Np} and use variational Bayes in place of the default stochastic expectation maximization algorithm. Parameters are inferred over 31 equally spaced intervals from 133 to 133016 generations in the past, and the sampled posterior ARGs are reported. 

The demographic model used to seed inference is given on page~\pageref{app:dem_model:seed}. We visualise this demographic model in the POPdemog package in Figure \ref{smc2demog} \cite{Zhou2018}.

Each {\tt SMCSMC} analysis gives a final output file detailing migration and coalescent events, their rates, and their opportunities which denote the total opportunity for an event to occur during a particular epoch. Output files are trimmed to only visualise the final epoch of variational Bayes inference and assessed for convergence. Times and rates are interpreted differently than {\tt scrm} output. Rates are in units of $4N_0$ per generation, while times are given in generations. 

We implement the above in a {\tt Snakemake} pipeline. Sample size and relative cross-coalescent rates are transformed as described in the documentation using the same parameter values for mutation rate and generation time used for {\tt SMCSMC} analysis. Effective population size and migration estimates for the populations analysed in the SGDP are given in Figures \ref{sgdp_ne} and \ref{sgdp_mig}. MSMC appears to consistently find a higher African $N_e$ in the ancient past until the average estimates across populations stabilises approximately 100kya (Figure \ref{fig:both}a). We expand on a possible reason for this effect in the main text of this article.


Migration during the last 100ky is integrated into a metric we call the inferred migration fraction (IMF). (Figure \ref{sgdp_heatmap}). This is related to the cummulative migration fraction (CMF) as introduced in MSMC-IM \cite{Wang2019a}, except the quantity is integrated in a particular epoch. We use two methods to integrate migration, the first presented in the main text given by $F(t) = e^{- \int_{t=0}^T \rho(t) dt}$. Alternatively, consider $p$ proportion of the population are replaced every generation. Start with 0 individuals from the source $N_{source}$ population in the sink population $N_{sink}$, each generation replace $p$ proportion of the sink population with the source. We track the proportion of the population which are replaced by the source $P$.  

$$ \begin{aligned} P_0 &= 0 \\ P_1 &= pN_{sink} \\ P_2 &= pN_{sink} + p(N_{sink} - pN_{sink}) \\ &= pN_{sink} + pN_{sink}(1-p) \\ P_3 &= pN_{sink} + pN_{sink}(1-p) + p((N_{sink}-pN_{sink}) - p(N_{sink}-pN_{sink})) \\ &= pN_{sink} + pN_{sink}(1-p)+p(N_{sink}(1-p)-pN_{sink}(1-p)) \\ &= pN_{sink}+pN_{sink}(1-p)+pN_{sink}(1-p)(1-p) \\ &\dots \\ P_n &=N_{sink}p(1-p)^n \end{aligned} $$

In practice, both methods give essentially identical proportions for all considered questions. Inferred migration varies across language groups (Figure \ref{migrationplot}, Table \ref{table:sgdp_pairwise}). Afroasiatic groups show high migration from Han and French populations, with a lower proportion deriving from Papuans. Niger-Kordofanian and Nilo-Saharan groups show an intermediate magnitude, between 50 and 60 percent replacement, though also significantly (P < 0.05, two-tailed paired $T$ test) closer to French and Han sources than Papuans. The Khoesan show the lowest migration, consistent with their early diversification from the remainder of African groups and the relative lack of gene-flow from Western African populations \cite{Lipson2019}. This is contrasted with the Mbuti and Biaka, Central African Hunter Gatherer populations who have historically received substantial amounts of gene flow from Western African sources. Both of these populations show the lowest migration in their language group (Table \ref{average_sgdp_migration_table}).  

We track the overall peak of migration rate in different populations (Figure \ref{fig:peaks}a,b). The most common backwards migration peak falls in the epoch between 35--45kya in the Nilo-Saharan and Niger-Kordofanian groups. Forwards migration has an earlier peak, in the epoch spanning 55--70kya. This result must be interpreted in light of the simulation results presented below.  

We model the migration adjusted $N_e$ in Eurasian populations, averaged over African partners, and African populations averaged over Eurasian partners \ref{fig:individual_pop_sizes}. The resulting curves largely represent our prior knowledge of world history, with an early divergence of Papuans consistent with the timing proposed in \cite{Malaspinas2016}, and a second bottleneck of populations inhabiting North America such as the Karitiana and Pima. Because we do not explicitly infer population split times, and have no convenient metric like {\tt MSMC2}, more fine-scale trends are difficult to identify. The African population size models show more discrepancy between populations, including an OoA-like bottleneck in Afroasiatic populations, and a large historical population size in hunter-gatherer groups such as proposed in \cite{Lipson2019}. 



\subsection{Validation in a physically phased subset of the Human Genome Diversity Panel (HGDP)} \label{hgdp_section}

Phased data is not essential for demographic inference using {\tt SMCSMC}; however, the use of phase alongside the look-ahead likelihood allows for more efficient convergence. The Human Genome Diversity Project collected 929 genomes from a diverse collection of human populations \cite{Bergstrom2019}. 36 of these genomes, two each from nine Eurasian and four African populations, were physically phased by use of linked-read sequencing technologies. This resource allows us to validate our inference both in an independent dataset, and evaluate the effect of phasing errors on {\tt SMCSMC} inference.   

To analyse these data, the same {\tt Snakemake} pipeline was used with minor adjustments in wildcard constrains to account for differences in sample names. 120 chunks of the genome were run in parallel for reasons of computational efficiency, while fixed recombination rate and mutation rates were held at the same values as the SGDP analysis, and an identical demographic model was used to initiate the analysis. Three replicates of the analysis were performed to assess the impact of stochastic sampling variation on inference. We infer both effective population size and directional migration in each of these 9x4 comparisons between Eurasian and African populations (Figure \ref{fig:both}a). The resulting inference allows us to verify and validate many observations from the SGDP.

Firstly, we calculate the timing of the migration peak and its magnitude, and find the estimates largely in line with the SGDP inference (Figure \ref{fig:peaks}c,d). For instance, inferred backwards migration in the Yoruban and Biaka populations peak at 40-50kya, while the Mbuti and San show earlier migration peaks around 50-60kya (Figure \ref{fig:peaks}c). The migration rate at the peak shows the same qualitative trends as the SGDP, with the peak in the Yoruban (approximately $2.5\times10^{-4}$) far exceeding the peaks in the Biaka, Mbuti, or San (between $0.1-0.175\times10^{-4}$) (Figure \ref{fig:peaks}d). This replication in the HGDP confirms the presence of a large directional migration in the Late Middle Pleistocene, and demonstrates that statistical errors in phasing the SGDP are not large contributors to the qualitative trends observed. 

We integrate migration between 40--70kya to obtain the inferred IMF for each of the comparisons in the HGDP (\ref{fig:both}b). Differences amung the African populations mirror those in the SGDP, though the proportions are uniformly smaller (main text). However, the migration rates backwards into Africa are apparent in all comparisons, and the order of populations IMF remains the same. Differences between individual Eurasian donor populations are small, and with the exception of the Papuan, insignificant. A discussion of the Papuan comparisons appears in the subsequent section.  

To compare the HGDP inference with the SGDP inference, we construct a set of the SGDP with the same donor populations as the HGDP.



%For the SMCSMC algorithm, the use of phased data is not necessary but does help with efficient convergence by assisting the lookahead-likelihood-guided resampling. Therefore, we do not expect errors made during statistical phasing to significantly impact the inferred parameters. However, to test this, we replicate the above analysis in a physically phased subset of the HGDP downlaoded from \url{ftp://ngs.sanger.ac.uk/production/hgdp/hgdp_wgs.20190516/}. The same {\tt snakemake} pipeline is used as in the analysis of the SGDP data. Data is additionally masked for the filters provided with the data. 

%We first asked if we could replicate the inflated African $N_e$ estimates from the main text in this new, high quality, data set. To do this, we infer effective population size with SMCSMC and MSMC.  The full inference from both algorithms is shown in \ref{fig:hgdp_ne} and \ref{fig:hgdp_mig}. For SMCSMC, we use 10,000 particles and 15 iterations to achieve convergence, with the same recombination and mutation rates as in the SGDP analysis.  The source populations here are more diverse than in the main text, comprising a sample of global diversity including Druze, Han, Karitiana, two Papuans, Pathan, Pima, Sardinians, and Yakut. To obtain a comparable sample from the SGDP, we select populations matching those in the HGDP. We match each of the source and sink populations, and similarly infer their effective population size and migration histories. The full inference is shown in Figure \ref{hgdp_sgdp_ne} and \ref{fig:hgdp_sgdp_mig}.


\subsection{Comparisons between the HGDP and a subset of the SGDP}

Previously, inference in the SGDP has relied on three candidate Eurasian donor populations. However, the physically phased subset of the HGDP provides a higher resolution view into global migration patterns with nine Eurasian populations represented. In order to compare effectively between the inferences made in these two datasets, we find representatives from these nine Eurasian populations in the SGDP dataset and use them as donor populations to the same four African populations (Yoruban, San, Mbuti, and Biaka), effectively recreating the analysis done in the physically phased subset of the HGDP. We select the Khomani San as a representative of the San, and only use one of the Papuan populations in the HGDP to compare (Highlands, as opposed to Sepik), creating the same 8x4 analysis table for both data sets. We infer the effective population sizes and migration rates using both {\tt SMCSMC} and {\tt MSMC}, with analysis details effectively identical to the original comparisons listed above (Figure \ref{fig:hgdp_sgdp}). We average over inferences to visually compare trends between the two datasets, in the same populations and compute the inferred IMF between 40--70kya (Figure \ref{fig:both}).  

In both the HGDP and the SGDP, MSMC estimates of African population size are higher than {\tt SMCSMC} estimates in the ancient past (80 -- 300kya) (Figure \ref{fig:both}a). By modelling directional migration, we are able to account for excess genetic diversity in the ancestral African population in both datasets. Uncertainty in the estimates increases substantially nearer to the present, as would be expected with the {\tt SMCSMC} method. 

We summarise migration from 40--70kya in the HGDP similarly to the SGDP. The total inferred migration is lower in the HGDP than in the SGDP (Figure \ref{fig:both}b). We use this comparison setup to additionally test the differences between Papuan donors and the remainder of Eurasians. We construct a linear model predicting IMF based on an indicator variable of Papuan/not Papuan and the receptor donor population, and find that in both the SGDP and the HGDP, Papuans show approximately 2\% less IMF than other donor populations (Tables \ref{hgdp:papuan_imf}, \ref{sgdp:papuan_imf}). While this difference is small, it is highly significant. However, the demographic scenario causing this difference in inferred IMF is not obvious; it is possible that the Papuan group had begun to diverge from the donating population prior to the admixture event, or alternatively that differences in archaic admixture between Eurasian and Papuan groups make up the difference in affinity. 



However, the qualitative patterns in inferred directional migration rates between populations are similar in both datasets (Figure \ref{fig:both}c). In both datasets, the highest rates are found in the Yorubans, follow by the Biaka, then the Mbuti and San. The MSMC curves are interestingly dissimilar between the different data sets, with a much steeper ascent around the period of our inferred migration in the HGDP than the SGDP. 


\begin{table}[ht]
    \centering
    \begin{tabular}{lll}
      \hline
    Name & ID & Source \\ 
      \hline
    French & S\_French-1 & SGDP \\ 
      Han & S\_Han-1 & SGDP \\ 
      Papuan & S\_Papuan-1 & SGDP \\ 
      BantuHerero & S\_BantuHerero-1 & SGDP \\ 
      BantuKenya & S\_BantuKenya-1 & SGDP \\ 
      BantuTswana & S\_BantuTswana-1 & SGDP \\ 
      Biaka & S\_Biaka-1 & SGDP \\ 
      Dinka & B\_Dinka-3 & SGDP \\ 
      Esan & S\_Esan-1 & SGDP \\ 
      Gambian & S\_Gambian-1 & SGDP \\ 
      Ju hoan North & S\_Ju\_hoan\_North-1 & SGDP \\ 
      Khomani San & S\_Khomani\_San-1 & SGDP \\ 
      Luhya & S\_Luhya-1 & SGDP \\ 
      Luo & S\_Luo-1 & SGDP \\ 
      Mandenka & S\_Mandenka-1 & SGDP \\ 
      Masai & S\_Masai-1 & SGDP \\ 
      Mbuti & S\_Mbuti-1 & SGDP \\ 
      Mende & S\_Mende-1 & SGDP \\ 
      Mozabite & S\_Mozabite-1 & SGDP \\ 
      Saharawi & S\_Saharawi-1 & SGDP \\ 
      Somali & S\_Somali-1 & SGDP \\ 
      Yoruba & S\_Yoruba-1 & SGDP \\ 
      Druze & HGDP00562 & HGDP \\ 
      Han & HGDP00774 & HGDP \\ 
      Karitiana & HGDP01013 & HGDP \\ 
      PapuanHighlands & HGDP00549 & HGDP \\ 
      PapuanSepik & HGDP00542 & HGDP \\ 
      Pathan & HGDP00224 & HGDP \\ 
      Pima & HGDP01043 & HGDP \\ 
      Sardinian & HGDP00670 & HGDP \\ 
      Yakut & HGDP00946 & HGDP \\ 
      Yoruba & HGDP00930 & HGDP \\ 
      San & HGDP01029 & HGDP \\ 
      Mbuti & HGDP00450 & HGDP \\ 
      Biaka & HGDP00460 & HGDP \\ 
      Vindija & Vindija.DG & Pruefer et al. 2017 \\ 
      Altai & Altai\_published.DG & Pruefer et al. 2013 \\ 
      Denisovan & Deniosva\_published.DG & Myers et al 2012 \\ 
       \hline
    \end{tabular}
    \caption{Sample IDs of the individuals used in this article and relevant resources. \gls{sgdp} = Simons Genome Diversity Project, \gls{hgdp} = Human Genome Diversity Panel.} 
    \label{samples}
    \end{table}
    

    \subsection{Statistical Analysis of Migrated Segments} \label{dstats_section}

    We run {\tt SMCSMC} with the {\tt -arg} flag to report the posterior estimate of the ancestral recombination graph. We use this to isolate segments of the African genome where predicted migration events occurred between 50 and 70kya and used these segments to calculate drift statistics. The isolation procedure is implemented in {\tt smcsmc.find\_segments}, and involves sequentially reconstructing marginal trees and keeping track of which contain migration events in a particular epoch. We isolate segments from the marginal trees of all SGDP comparisons. 

\subsection{Length Distribution of Isolated Segments}


 Under the Markovian model of the SMC', the length of admixed tracts $L$ is an exponential process with scale factor $2N (1 - m ) \left( 1 - e^{-T / 2N} \right)$, with a proportion $m$ of the sink population being replaced with the source $T$ generations in the past and an effective population size of $N$ \cite{Marjoram2006,Liang953}. This gives an approximate mean length $\left[ (1 -m)r(T-1) \right]^{-1}$ with recombination rate $r$ in units of Morgans, which is well approximated
by  $(rT)^{-1}(1-m)$ \cite{Racimo2015}; we use this approximation to derive expected distribution of fragment sizes. When analysing populations with {\tt SMCSMC}, we fix the recombination rate at $3 \times 10^{-9}$ uniformly across the genome, in line with that used by {tt MSMC} in simulations \cite[Supp.\ section 7]{Schiffels2014}.
This value is a conservative underestimate, accounting for the presence of recombination hotspots and {\tt SMCSMC}'s inability to deconvolve recombinations in these areas, effectively underestimating the true $r$.  For estimates of ancestral tract lengths, we use the more universally accepted value of $1 \times 10^{-8}$, equivalent to a one percent chance of a cross-over per megabase and per generation \cite{Dumont2008}. 
%\todo{better reference for recombination rate value}

We take the isolated segments (Figure \ref{fig:length}a) and compute the mean track length (Table \ref{table:lengths}). We use the approximation that the mean segment length should be approximately equal to $((1-m)r(T-1))^{-1}$ to determine that, if the migration happened in one pulse, our empirical distribution would suggest either a recent timing or a very large pulse (Figure \ref{fig:length}b). However, we heavily caveat any interpretation of these data with the fact that they are explicitly generated under a model of a given migration proportion. The fact, therefore, that they are of a consistent length with a large migration is more evidence for the model producing internally-consistent tracts than any external validation of the results in this article. 

The assumption that migration has occurred in a single wave is largely unrealistic. We used expectation maximisation to investigate if a mixture of exponential distributions explained the observed tract lengths better than a single distribution. We found that in some cases, two or three exponential distributions were better supposed by the data, however the differences in log likelihoods was negligible and the support for the different distributions was approximately inversely proportional to their number (data not shown). We found no strong support for multiple waves of migration from this analysis.

\subsection{Drift Statistics}

Patterson's formal statistics were calculated with {\tt ADMIXTOOLS} \cite{Patterson2012} and the {\tt admixr} package \cite{Petr2019} in {\tt R}. We converted the above sequence data to Eigenstrat format with {\tt vcf2eigenstrat} %\url{https://github.com/bodkan/vcf2eigenstrat} 
formerly distributed with {\tt admixr}. We merged SGDP and archaic Eigenstrat datasets with {\tt convertf} and {\tt mergeit} implemented in {\tt ADMIXTOOLS}. 

Here, Yoruba-1 is used as a representative of Western African groups, and used for ascertaining putatively migrated segments. Yoruba-2 is used as a comparison individual from the same populations. In this way, we look for evidence above another individual in the same population of similarity to Eurasians.

We first use $f_3$ statistics to look for evidence of admixture between the African and various Eurasian groups. We calculate $f_3$(Yoruba-1, Eurasian group, Yoruba-2) for Papuans, French, Han Chinese, and the Vindija Neanderthal. We calculate this statistic in all available markers, and additionally for the segments isolated from the three Eurasians separately (Figure \ref{fig:f3}). These statistics show, firstly, that ascertaining in a particular group influences the shared drift with that group. This is exemplified by the non-significant shared drift with Papuans in French and Han ascertained segments. Secondly, these statistic show significant levels ($|Z|>2$) of drift between the test individual and Eurasian populations, while also showing no increase in Neanderthal allele sharing ($f_3$ = 0). To find statistical evidence of admixture, we compute $f_3$(Yoruba-2, Eurasian, Yoruba-1) for the same Eurasian groups. We find statistical evidence for admixture in each of the groups examined, for all ascertainment schemes (Figure \ref{fig:f3_admix}). 

We use $D$ statistics to examine more nuanced scenarios. We find that the two Yorubans share more alleles than other groups in Africa (D(African group, Yoruba-1; Yoruba-2, Chimp) is significantly negative with $|Z|>3$), but the individual of interest is closer to Out of Africa (OoA) groups such as the Han, French, and Papauns (D(OoA, Yoruba-2; Yoruba-1, Chimp) is significantly negative with $|Z|<3$) than to its partner Yoruban (Table \ref{dstats:a1}). This implies that {\tt SMCSMC} has identified segments of the African Genome which are more closely related to OoA populations than to fellow Africans.   


\subsection{Simulation procedure} \label{simproc}

Coalescent simulations were performed under the sequential coalescent with recombination model ({\tt SCRM}) \cite{Staab2015}. 1 gigabase (Gb) of sequence was simulated.  In addition to branches in local genealogical trees, {\tt SCRM} retains non-local branches in the ancestral recombination graph (ARG) within a user-specified sliding window.  In the limit of a chromosome-sized windows {\tt SCRM} is equivalent to the coalescent with recombination, while for a zero-length window it is equivalent to the sequentially Markovian coalescent (SMC') \cite{McVean2005,Marjoram2006}; we use a $100$kb sliding window to approximate the CwR and improve accuracy over SMC' while retaining tractable inference.

We modelled migration as a 10ky pulse of constant migration rate resulting in an integrated migration fraction (IMF) of 0 to 0.593. The migration pulse was centered at various times between 40 and 70 kya.  Due to the amount of compute required, we then used {\tt SMCSMC} to infer the demographic parameters using a reduced set of 5000 particles and 5 iterations of the VB procedure. To aid convergence, we started inference at a reasonable approximation of human demographic history (see Supplemental Section \ref{simproc}.1, Supplemental Fig.\ \ref{fig:dem}). We modelled $N_e$ and migration rates as piecewise continuous functions and set 32 exponentially spaced epochs from 133 to 133016 generations in the past. To convert evolutionary rates to years we set a generation time of 29 years \cite{Fenner2005}.  For computational efficiency, individual genomes were split into 120 chunks and processed in parallel, with sufficient statistics collected and processed together in the VB steps.

Times are in units of $4N_0g$ while population sizes are in units of $N_0$. For $g=29, N_0 = 14312$, the demographic model is as shown in Figure \ref{fig:dem}.  Exact specifications are given in \ref{app:dem_model}

The demographic model which we have assumed for both population's effective sizes has been shown to recapitulate similar inference to real data (data not shown). The migration parameter must be initiated at a given magnitude; further back in time, the particle filter is less able to identify lineage's true populations, and the inference of migration rates becomes essential uniform. Thus, we see a ``drop-off'' effect, where in the ancient past, the inference remains at the initiation value, and as more certainty about different histories is obtained, the migration values recapitulate real information. Thus the choice of an appropriate parameter for the initial migration rate is a crucial step in {\tt SMCSMC} analysis, and here we chose to arrive at this value through simulation.

We simulate back-migration scenarios of varying total migration proportions from 0 (no migration) up to 60\% population replacement. For each simulation, we initiate the particle filter at either 0, 1, or 5 4$N_0$ proportion replaced per generation (which are the units used internally by {\tt scrm} and {\tt ms} for simulation).  {\tt SMCSMC} is then used to infer effective population size and migration histories in five iterations with 5000 particles. As a cautionary note, these simulations are almost certainly not fully converged, and are used as an indication of power. Their power, theoretically, approaches 1, as particle filters asymptotically exactly approach the true posterior distribution. However, these low resolution attempts are indicative of a ``quick'' overview of the abilities of the algorithm. With 600 cores available, each of the cases (forward, backward, or bidirectional) was able to run in approximately 20 hours. 

Generally, beginning with a higher migration rate seems to recover a higher proportion of the simulated migration. However, as in the case of a 60\% replacement simulated 40kya, beginning with 5 4$N_0$ rather than 1 4$N_0$ recovers similar proportions of backwards migration (0.502 vs 0.52) yet the higher migration rate finds 0.301 Eurasian migration rather than 0.195. The higher initial migration rates thus slightly reduce power (though, not in all cases, and for fully converged solutions, we would expect both proportions to be similar up to noise) while additionally finding an increased migration in the opposite direction.  Beginning with a zero rate leads to highly unstable estimates of the migration rate and effective population size, and we exclude it from our analysis.

We select a more comprehensive set of initiation parameters and particle values and use them to analyze a Yoruban and French individual from SGDP (Fig \ref{init_yri}). The effect of the initial migration rate seems relatively consistent for low values (0.5 - 2.0),while an increasingly small migration peak is seen for higher initial magnitudes 4.0 - 10.0. Again, beginning with an initial rate of zero tends to lead to highly unstable estimates of effective population size and migration rates. For the remainder of the analyses in this article, we choose to use an initial rate of 1.0. 



\subsection{Isolating Anciently Admixed Segments} We sampled genealogical trees with migration events from the posterior distribution estimated by the particle filter under the final, converged, demographic parameters. We scan along the sequence and identified marginal trees with migration events from the source (Eurasian) population to the sink (African) population (forward in time) within the desired time period along with the beginning and end position of that tree in the genome sequence. In this process, we ignore recombination events that alter a tree in such a way that the migration event is retained.  

\subsection{Sequence Data and Preparation}  We downloaded whole genome sequence (WGS) data from the phased release of the Simons Genome Diversity Panel and converted it to {\tt .seg} file format using scripts provided (See URLs). We apply two masks to the data. First, we mask the data with the strict accessibility mask provided by the 1000 genomes project (see URLs). Second, we mask any sites absent chimpanzee ancestry, to address a known variant issue in the data that resulted in artificially long runs of homozygosity \cite{Wang2019a}. We develop a {\tt Snakemake} \cite{Koster2012} pipeline for efficiently analysing sequence data with both {\tt SMCSMC} and {\tt MSMC2}. We assume a mutation rate of $1.25\times10^{-8}$ and a recombination rate of $3\times10^{-9}$ (events per nucleotide per generation), in line with recent literature \cite{Scally2012, Schiffels2014a}. The number of particles, and the number of VB iterations, are set per analyses, and are reported in figure captions. Unless otherwise noted, the names of individuals used in this paper are the first in their population (e.g.\ an individual named Yoruban is {\tt S\_Yoruba-1} in the SGDP nomenclature); a complete list of sample identifiers is provided in Supplemental Table \ref{samples}. 



\subsection{Integrated Migration Fraction}
The IMF, the total fraction of a particular population $A$ replaced during a particular time period from $T_0$ to $T_1$ generations in the past is found as follows. Let $\rho(t)$ be the instantaneous rate of migration out of $A$ per unit of time in the backward direction (i.e.\ into $A$ forwards in time), and $F(t)$ the fraction not migrated in the epoch $[T_0,t]$, then $\frac{d}{dt} F(t) = -\rho(t) F(t)$ with solution $F(t) = e^{- \int_{T_0}^{T_1} \rho(t) {\mathrm{d}t}}$, so that the IMF is given by $1-F(T_1)$.  The integral is calculated as a finite sum since $\rho$ is piecewise constant.
